{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "c77f8da1-2837-4bf7-b3f4-dfcaa7ab42f6",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-11-08T16:07:38.861414Z",
     "iopub.status.busy": "2023-11-08T16:07:38.861252Z",
     "iopub.status.idle": "2023-11-08T16:07:38.964903Z",
     "shell.execute_reply": "2023-11-08T16:07:38.964298Z"
    }
   },
   "outputs": [],
   "source": [
    "import scipy.io\n",
    "import numpy as np\n",
    "import random\n",
    "import os"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "3fe25376-b856-4296-ba02-5e3771ff9451",
   "metadata": {
    "collapsed": true,
    "execution": {
     "iopub.execute_input": "2023-11-08T16:07:38.967314Z",
     "iopub.status.busy": "2023-11-08T16:07:38.967012Z",
     "iopub.status.idle": "2023-11-08T17:59:05.798270Z",
     "shell.execute_reply": "2023-11-08T17:59:05.797727Z"
    },
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 11:07:39.146083: I tensorflow/core/platform/cpu_feature_guard.cc:182] This TensorFlow binary is optimized to use available CPU instructions in performance-critical operations.\n",
      "To enable the following instructions: FMA, in other operations, rebuild TensorFlow with the appropriate compiler flags.\n",
      "/usr/lib/python3.11/site-packages/h5py/__init__.py:36: UserWarning: h5py is running against HDF5 1.14.2 when it was built against 1.14.1, this may cause problems\n",
      "  _warn((\"h5py is running against HDF5 {0} when it was built against {1}, \"\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "35512\n",
      "WARNING:tensorflow:From /usr/lib/python3.11/site-packages/tensorflow/python/compat/v2_compat.py:107: disable_resource_variables (from tensorflow.python.ops.variable_scope) is deprecated and will be removed in a future version.\n",
      "Instructions for updating:\n",
      "non-resource variables are not supported in the long term\n",
      "WARNING:tensorflow:Layer lstm will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_1 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_2 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_3 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_4 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 11:09:05.732535: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 11:09:05.744552: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 11:09:05.744796: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 11:09:05.748288: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 11:09:05.748502: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 11:09:05.748696: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 11:09:05.843766: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 11:09:05.844057: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 11:09:05.844336: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 11:09:05.844522: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1639] Created device /job:localhost/replica:0/task:0/device:GPU:0 with 46608 MB memory:  -> device: 0, name: NVIDIA RTX A6000, pci bus id: 0000:2d:00.0, compute capability: 8.6\n",
      "2023-11-08 11:09:05.844851: I tensorflow/core/common_runtime/process_util.cc:146] Creating new thread pool with default inter op setting: 2. Tune using inter_op_parallelism_threads for best performance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WARNING:tensorflow:Layer lstm_5 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_6 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_7 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_8 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_9 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_10 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_11 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_12 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_13 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_14 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_15 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_16 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_17 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_18 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_19 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_20 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_21 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 11:09:06.795471: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 11:09:36.132067: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 11:09:36.132369: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 11:09:36.132603: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 11:09:36.132798: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 11:09:36.133002: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1639] Created device /job:localhost/replica:0/task:0/device:GPU:0 with 46608 MB memory:  -> device: 0, name: NVIDIA RTX A6000, pci bus id: 0000:2d:00.0, compute capability: 8.6\n",
      "2023-11-08 11:09:36.133047: I tensorflow/core/common_runtime/process_util.cc:146] Creating new thread pool with default inter op setting: 2. Tune using inter_op_parallelism_threads for best performance.\n",
      "2023-11-08 11:09:36.156385: I tensorflow/compiler/mlir/mlir_graph_optimization_pass.cc:375] MLIR V1 optimization pass is not enabled\n",
      "2023-11-08 11:09:36.391970: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_21/lstm_cell_21/bias/Assign' id:3510 op device:{requested: '', assigned: ''} def:{{{node lstm_21/lstm_cell_21/bias/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_21/lstm_cell_21/bias, lstm_21/lstm_cell_21/bias/Initializer/concat)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 11:09:36.524514: W tensorflow/c/c_api.cc:304] Operation '{name:'strided_slice/stack_1' id:3634 op device:{requested: '', assigned: ''} def:{{{node strided_slice/stack_1}} = Const[_has_manual_control_dependencies=true, dtype=DT_INT32, value=Tensor<type: int32 shape: [4] values: 0 0 0...>]()}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 11:09:36.557796: W tensorflow/c/c_api.cc:304] Operation '{name:'strided_slice/stack_2' id:3635 op device:{requested: '', assigned: ''} def:{{{node strided_slice/stack_2}} = Const[_has_manual_control_dependencies=true, dtype=DT_INT32, value=Tensor<type: int32 shape: [4] values: 1 1 1...>]()}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WARNING:tensorflow:Layer lstm_22 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_23 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_24 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_25 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_26 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_27 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_28 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_29 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_30 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_31 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_32 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_33 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_34 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_35 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_36 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "(31949, 95)\n",
      "Train on 31949 samples, validate on 3563 samples\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 11:09:40.976270: W tensorflow/c/c_api.cc:304] Operation '{name:'training/Adam/lstm_25/lstm_cell_25/kernel/v/Assign' id:17412 op device:{requested: '', assigned: ''} def:{{{node training/Adam/lstm_25/lstm_cell_25/kernel/v/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](training/Adam/lstm_25/lstm_cell_25/kernel/v, training/Adam/lstm_25/lstm_cell_25/kernel/v/Initializer/zeros)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Epoch 1/50\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 11:09:45.314608: E tensorflow/core/grappler/optimizers/meta_optimizer.cc:954] layout failed: INVALID_ARGUMENT: Size of values 0 does not match size of permutation 4 @ fanin shape indropout/cond/dropout/SelectV2-2-TransposeNHWCToNCHW-LayoutOptimizer\n",
      "2023-11-08 11:09:47.883135: I tensorflow/compiler/xla/stream_executor/cuda/cuda_blas.cc:606] TensorFloat-32 will be used for the matrix multiplication. This will only be logged once.\n",
      "2023-11-08 11:09:47.900608: I tensorflow/compiler/xla/stream_executor/cuda/cuda_dnn.cc:432] Loaded cuDNN version 8902\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31949/31949 [==============================] - ETA: 0s - loss: 3.2006"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/lib/python3.11/site-packages/keras/engine/training_v1.py:2335: UserWarning: `Model.state_updates` will be removed in a future version. This property should not be used in TensorFlow 2.0, as `updates` are applied automatically.\n",
      "  updates = self.state_updates\n",
      "2023-11-08 11:10:08.848680: W tensorflow/c/c_api.cc:304] Operation '{name:'loss_1/mul' id:6477 op device:{requested: '', assigned: ''} def:{{{node loss_1/mul}} = Mul[T=DT_FLOAT, _has_manual_control_dependencies=true](loss_1/mul/x, loss_1/dense_3_loss/value)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Epoch 1: val_loss improved from inf to 1.87679, saving model to ./checkpoints/unknown_person_baseline_p9_20.h5\n",
      "31949/31949 [==============================] - 29s 912us/sample - loss: 3.2006 - val_loss: 1.8768\n",
      "Epoch 2/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.7631\n",
      "Epoch 2: val_loss improved from 1.87679 to 1.56513, saving model to ./checkpoints/unknown_person_baseline_p9_20.h5\n",
      "31949/31949 [==============================] - 23s 713us/sample - loss: 1.7631 - val_loss: 1.5651\n",
      "Epoch 3/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.5761\n",
      "Epoch 3: val_loss improved from 1.56513 to 1.49002, saving model to ./checkpoints/unknown_person_baseline_p9_20.h5\n",
      "31949/31949 [==============================] - 21s 646us/sample - loss: 1.5761 - val_loss: 1.4900\n",
      "Epoch 4/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.5213\n",
      "Epoch 4: val_loss improved from 1.49002 to 1.45558, saving model to ./checkpoints/unknown_person_baseline_p9_20.h5\n",
      "31949/31949 [==============================] - 20s 630us/sample - loss: 1.5213 - val_loss: 1.4556\n",
      "Epoch 5/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.4984\n",
      "Epoch 5: val_loss improved from 1.45558 to 1.45360, saving model to ./checkpoints/unknown_person_baseline_p9_20.h5\n",
      "31949/31949 [==============================] - 20s 628us/sample - loss: 1.4984 - val_loss: 1.4536\n",
      "Epoch 6/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.4830\n",
      "Epoch 6: val_loss improved from 1.45360 to 1.42040, saving model to ./checkpoints/unknown_person_baseline_p9_20.h5\n",
      "31949/31949 [==============================] - 20s 625us/sample - loss: 1.4830 - val_loss: 1.4204\n",
      "Epoch 7/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.4718\n",
      "Epoch 7: val_loss improved from 1.42040 to 1.41136, saving model to ./checkpoints/unknown_person_baseline_p9_20.h5\n",
      "31949/31949 [==============================] - 20s 637us/sample - loss: 1.4718 - val_loss: 1.4114\n",
      "Epoch 8/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.4731\n",
      "Epoch 8: val_loss improved from 1.41136 to 1.40569, saving model to ./checkpoints/unknown_person_baseline_p9_20.h5\n",
      "31949/31949 [==============================] - 20s 632us/sample - loss: 1.4731 - val_loss: 1.4057\n",
      "Epoch 9/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.4717\n",
      "Epoch 9: val_loss improved from 1.40569 to 1.40211, saving model to ./checkpoints/unknown_person_baseline_p9_20.h5\n",
      "31949/31949 [==============================] - 20s 636us/sample - loss: 1.4717 - val_loss: 1.4021\n",
      "Epoch 10/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.4585\n",
      "Epoch 10: val_loss did not improve from 1.40211\n",
      "31949/31949 [==============================] - 20s 623us/sample - loss: 1.4585 - val_loss: 1.4041\n",
      "Epoch 11/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.4529\n",
      "Epoch 11: val_loss did not improve from 1.40211\n",
      "31949/31949 [==============================] - 20s 622us/sample - loss: 1.4529 - val_loss: 1.4101\n",
      "Epoch 12/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.4520\n",
      "Epoch 12: val_loss improved from 1.40211 to 1.38423, saving model to ./checkpoints/unknown_person_baseline_p9_20.h5\n",
      "31949/31949 [==============================] - 20s 631us/sample - loss: 1.4520 - val_loss: 1.3842\n",
      "Epoch 13/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.4475\n",
      "Epoch 13: val_loss did not improve from 1.38423\n",
      "31949/31949 [==============================] - 20s 625us/sample - loss: 1.4475 - val_loss: 1.4073\n",
      "Epoch 14/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.4427\n",
      "Epoch 14: val_loss improved from 1.38423 to 1.38412, saving model to ./checkpoints/unknown_person_baseline_p9_20.h5\n",
      "31949/31949 [==============================] - 20s 628us/sample - loss: 1.4427 - val_loss: 1.3841\n",
      "Epoch 15/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.5312\n",
      "Epoch 15: val_loss did not improve from 1.38412\n",
      "31949/31949 [==============================] - 21s 668us/sample - loss: 1.5312 - val_loss: 1.3934\n",
      "Epoch 16/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.6106\n",
      "Epoch 16: val_loss did not improve from 1.38412\n",
      "31949/31949 [==============================] - 22s 699us/sample - loss: 1.6106 - val_loss: 1.4280\n",
      "Epoch 17/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.4767\n",
      "Epoch 17: val_loss did not improve from 1.38412\n",
      "31949/31949 [==============================] - 20s 621us/sample - loss: 1.4767 - val_loss: 1.4069\n",
      "Epoch 18/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.4500\n",
      "Epoch 18: val_loss did not improve from 1.38412\n",
      "31949/31949 [==============================] - 21s 661us/sample - loss: 1.4500 - val_loss: 1.3860\n",
      "Epoch 19/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.4510\n",
      "Epoch 19: val_loss did not improve from 1.38412\n",
      "31949/31949 [==============================] - 23s 706us/sample - loss: 1.4510 - val_loss: 1.3866\n",
      "Epoch 20/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.5194\n",
      "Epoch 20: val_loss did not improve from 1.38412\n",
      "31949/31949 [==============================] - 23s 704us/sample - loss: 1.5194 - val_loss: 1.4061\n",
      "Epoch 21/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.5030\n",
      "Epoch 21: val_loss did not improve from 1.38412\n",
      "31949/31949 [==============================] - 23s 708us/sample - loss: 1.5030 - val_loss: 1.3910\n",
      "Epoch 22/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.5788\n",
      "Epoch 22: val_loss did not improve from 1.38412\n",
      "31949/31949 [==============================] - 20s 629us/sample - loss: 1.5788 - val_loss: 1.4360\n",
      "Epoch 23/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.4564\n",
      "Epoch 23: val_loss did not improve from 1.38412\n",
      "31949/31949 [==============================] - 21s 644us/sample - loss: 1.4564 - val_loss: 1.4012\n",
      "Epoch 24/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.4358\n",
      "Epoch 24: val_loss improved from 1.38412 to 1.38084, saving model to ./checkpoints/unknown_person_baseline_p9_20.h5\n",
      "31949/31949 [==============================] - 23s 707us/sample - loss: 1.4358 - val_loss: 1.3808\n",
      "Epoch 25/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.4296\n",
      "Epoch 25: val_loss did not improve from 1.38084\n",
      "31949/31949 [==============================] - 20s 621us/sample - loss: 1.4296 - val_loss: 1.3905\n",
      "Epoch 26/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.4243\n",
      "Epoch 26: val_loss improved from 1.38084 to 1.37564, saving model to ./checkpoints/unknown_person_baseline_p9_20.h5\n",
      "31949/31949 [==============================] - 20s 619us/sample - loss: 1.4243 - val_loss: 1.3756\n",
      "Epoch 27/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.4244\n",
      "Epoch 27: val_loss did not improve from 1.37564\n",
      "31949/31949 [==============================] - 22s 696us/sample - loss: 1.4244 - val_loss: 1.3796\n",
      "Epoch 28/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.4199\n",
      "Epoch 28: val_loss did not improve from 1.37564\n",
      "31949/31949 [==============================] - 23s 708us/sample - loss: 1.4199 - val_loss: 1.3780\n",
      "Epoch 29/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.4168\n",
      "Epoch 29: val_loss improved from 1.37564 to 1.37364, saving model to ./checkpoints/unknown_person_baseline_p9_20.h5\n",
      "31949/31949 [==============================] - 23s 706us/sample - loss: 1.4168 - val_loss: 1.3736\n",
      "Epoch 30/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.4119\n",
      "Epoch 30: val_loss improved from 1.37364 to 1.37310, saving model to ./checkpoints/unknown_person_baseline_p9_20.h5\n",
      "31949/31949 [==============================] - 21s 651us/sample - loss: 1.4119 - val_loss: 1.3731\n",
      "Epoch 31/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.4137\n",
      "Epoch 31: val_loss did not improve from 1.37310\n",
      "31949/31949 [==============================] - 23s 707us/sample - loss: 1.4137 - val_loss: 1.3766\n",
      "Epoch 32/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.4088\n",
      "Epoch 32: val_loss did not improve from 1.37310\n",
      "31949/31949 [==============================] - 22s 684us/sample - loss: 1.4088 - val_loss: 1.3750\n",
      "Epoch 33/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.4076\n",
      "Epoch 33: val_loss did not improve from 1.37310\n",
      "31949/31949 [==============================] - 22s 677us/sample - loss: 1.4076 - val_loss: 1.3759\n",
      "Epoch 34/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.4046\n",
      "Epoch 34: val_loss improved from 1.37310 to 1.37070, saving model to ./checkpoints/unknown_person_baseline_p9_20.h5\n",
      "31949/31949 [==============================] - 23s 709us/sample - loss: 1.4046 - val_loss: 1.3707\n",
      "Epoch 35/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.4040\n",
      "Epoch 35: val_loss did not improve from 1.37070\n",
      "31949/31949 [==============================] - 20s 636us/sample - loss: 1.4040 - val_loss: 1.3743\n",
      "Epoch 36/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.4010\n",
      "Epoch 36: val_loss improved from 1.37070 to 1.37058, saving model to ./checkpoints/unknown_person_baseline_p9_20.h5\n",
      "31949/31949 [==============================] - 20s 611us/sample - loss: 1.4010 - val_loss: 1.3706\n",
      "Epoch 37/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.4012\n",
      "Epoch 37: val_loss improved from 1.37058 to 1.35983, saving model to ./checkpoints/unknown_person_baseline_p9_20.h5\n",
      "31949/31949 [==============================] - 21s 643us/sample - loss: 1.4012 - val_loss: 1.3598\n",
      "Epoch 38/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3956\n",
      "Epoch 38: val_loss improved from 1.35983 to 1.35978, saving model to ./checkpoints/unknown_person_baseline_p9_20.h5\n",
      "31949/31949 [==============================] - 22s 692us/sample - loss: 1.3956 - val_loss: 1.3598\n",
      "Epoch 39/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3950\n",
      "Epoch 39: val_loss did not improve from 1.35978\n",
      "31949/31949 [==============================] - 20s 626us/sample - loss: 1.3950 - val_loss: 1.3629\n",
      "Epoch 40/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3927\n",
      "Epoch 40: val_loss did not improve from 1.35978\n",
      "31949/31949 [==============================] - 20s 620us/sample - loss: 1.3927 - val_loss: 1.3623\n",
      "Epoch 41/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3905\n",
      "Epoch 41: val_loss did not improve from 1.35978\n",
      "31949/31949 [==============================] - 21s 662us/sample - loss: 1.3905 - val_loss: 1.3685\n",
      "Epoch 42/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3895\n",
      "Epoch 42: val_loss improved from 1.35978 to 1.35642, saving model to ./checkpoints/unknown_person_baseline_p9_20.h5\n",
      "31949/31949 [==============================] - 23s 715us/sample - loss: 1.3895 - val_loss: 1.3564\n",
      "Epoch 43/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3895\n",
      "Epoch 43: val_loss did not improve from 1.35642\n",
      "31949/31949 [==============================] - 23s 711us/sample - loss: 1.3895 - val_loss: 1.3686\n",
      "Epoch 44/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3886\n",
      "Epoch 44: val_loss did not improve from 1.35642\n",
      "31949/31949 [==============================] - 23s 711us/sample - loss: 1.3886 - val_loss: 1.3566\n",
      "Epoch 45/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3856\n",
      "Epoch 45: val_loss did not improve from 1.35642\n",
      "31949/31949 [==============================] - 23s 708us/sample - loss: 1.3856 - val_loss: 1.3588\n",
      "Epoch 46/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3843\n",
      "Epoch 46: val_loss improved from 1.35642 to 1.35048, saving model to ./checkpoints/unknown_person_baseline_p9_20.h5\n",
      "31949/31949 [==============================] - 23s 713us/sample - loss: 1.3843 - val_loss: 1.3505\n",
      "Epoch 47/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3844\n",
      "Epoch 47: val_loss did not improve from 1.35048\n",
      "31949/31949 [==============================] - 23s 704us/sample - loss: 1.3844 - val_loss: 1.3551\n",
      "Epoch 48/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3837\n",
      "Epoch 48: val_loss did not improve from 1.35048\n",
      "31949/31949 [==============================] - 22s 693us/sample - loss: 1.3837 - val_loss: 1.3606\n",
      "Epoch 49/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3790\n",
      "Epoch 49: val_loss did not improve from 1.35048\n",
      "31949/31949 [==============================] - 23s 706us/sample - loss: 1.3790 - val_loss: 1.3588\n",
      "Epoch 50/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3781\n",
      "Epoch 50: val_loss improved from 1.35048 to 1.34964, saving model to ./checkpoints/unknown_person_baseline_p9_20.h5\n",
      "31949/31949 [==============================] - 23s 716us/sample - loss: 1.3781 - val_loss: 1.3496\n",
      "WARNING:tensorflow:From /usr/lib/python3.11/site-packages/tensorflow/python/ops/init_ops.py:94: calling GlorotUniform.__init__ (from tensorflow.python.ops.init_ops) with dtype is deprecated and will be removed in a future version.\n",
      "Instructions for updating:\n",
      "Call initializer instance with the dtype argument instead of passing it to the constructor\n",
      "WARNING:tensorflow:From /usr/lib/python3.11/site-packages/tensorflow/python/ops/init_ops.py:94: calling Zeros.__init__ (from tensorflow.python.ops.init_ops) with dtype is deprecated and will be removed in a future version.\n",
      "Instructions for updating:\n",
      "Call initializer instance with the dtype argument instead of passing it to the constructor\n",
      "WARNING:tensorflow:From /usr/lib/python3.11/site-packages/tensorflow/python/ops/init_ops.py:94: calling Orthogonal.__init__ (from tensorflow.python.ops.init_ops) with dtype is deprecated and will be removed in a future version.\n",
      "Instructions for updating:\n",
      "Call initializer instance with the dtype argument instead of passing it to the constructor\n",
      "WARNING:tensorflow:Layer lstm will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_1 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_2 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_3 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_4 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_5 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_6 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_7 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_8 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_9 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_10 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_11 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_12 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_13 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_14 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_15 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_16 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_17 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_18 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_19 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_20 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_21 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_22 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_23 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_24 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_25 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_26 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_27 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_28 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_29 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_30 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_31 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_32 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_33 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_34 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_35 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_36 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 11:27:39.258929: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_3_1/lstm_cell_40/kernel/Assign' id:19120 op device:{requested: '', assigned: ''} def:{{{node lstm_3_1/lstm_cell_40/kernel/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_3_1/lstm_cell_40/kernel, lstm_3_1/lstm_cell_40/kernel/Initializer/random_uniform)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 11:27:40.955826: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_29_1/lstm_cell_66/recurrent_kernel/v/Assign' id:26335 op device:{requested: '', assigned: ''} def:{{{node lstm_29_1/lstm_cell_66/recurrent_kernel/v/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_29_1/lstm_cell_66/recurrent_kernel/v, lstm_29_1/lstm_cell_66/recurrent_kernel/v/Initializer/zeros)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "/usr/lib/python3.11/site-packages/keras/engine/training_v1.py:2359: UserWarning: `Model.state_updates` will be removed in a future version. This property should not be used in TensorFlow 2.0, as `updates` are applied automatically.\n",
      "  updates=self.state_updates,\n",
      "2023-11-08 11:27:42.285307: W tensorflow/c/c_api.cc:304] Operation '{name:'dropout_4_1/cond/Merge' id:24591 op device:{requested: '', assigned: ''} def:{{{node dropout_4_1/cond/Merge}} = Merge[N=2, T=DT_FLOAT, _has_manual_control_dependencies=true](dropout_4_1/cond/Identity, dropout_4_1/cond/dropout/SelectV2)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1485, 1489)\n",
      "(1514, 1489)\n",
      "(1644, 1489)\n",
      "(1764, 1489)\n",
      "(1836, 1489)\n",
      "(1699, 1489)\n",
      "(1826, 1489)\n",
      "(1594, 1489)\n",
      "(1764, 1489)\n",
      "(1490, 1489)\n",
      "(1944, 1489)\n",
      "(1727, 1489)\n",
      "(1776, 1489)\n",
      "(1836, 1489)\n",
      "(1740, 1489)\n",
      "(1812, 1489)\n",
      "(946, 1489)\n",
      "(1692, 1489)\n",
      "(1860, 1489)\n",
      "{1: 6.000262902653801, 2: 8.150015249112112, 4: 10.0, 5: 4.03658308434664, 6: 4.134882140190613, 8: 9.037328764981138, 10: 6.503991400600146, 11: 7.979224435275812, 12: 8.64277462191395, 13: 6.236066680604741, 17: 9.402697870261047, 19: 8.060228780277638, 21: 8.333941718253651, 22: 1.0, 25: 7.578464121559212, 26: 6.36539488376651, 27: 7.803208360114843, 28: 5.214158757947024, 29: 1.4772533287892433}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_2791896/18730893.py:300: DeprecationWarning: Conversion of an array with ndim > 0 to a scalar is deprecated, and will error in future. Ensure you extract a single element from your array before performing this operation. (Deprecated NumPy 1.25.)\n",
      "  w_train[i] = weights_dict[int(p_train[i])]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Train on 31949 samples, validate on 3563 samples\n",
      "Epoch 1/20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 11:31:25.537167: E tensorflow/core/grappler/optimizers/meta_optimizer.cc:954] layout failed: INVALID_ARGUMENT: Size of values 0 does not match size of permutation 4 @ fanin shape indropout_5/cond/dropout/SelectV2-2-TransposeNHWCToNCHW-LayoutOptimizer\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31949/31949 [==============================] - ETA: 0s - loss: 9.3464\n",
      "Epoch 1: val_loss improved from inf to 1.38352, saving model to ./checkpoints/unknown_person_p9_20.h5\n",
      "31949/31949 [==============================] - 27s 846us/sample - loss: 9.3464 - val_loss: 1.3835\n",
      "Epoch 2/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 9.2870\n",
      "Epoch 2: val_loss improved from 1.38352 to 1.37473, saving model to ./checkpoints/unknown_person_p9_20.h5\n",
      "31949/31949 [==============================] - 22s 701us/sample - loss: 9.2870 - val_loss: 1.3747\n",
      "Epoch 3/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 9.2222\n",
      "Epoch 3: val_loss improved from 1.37473 to 1.35966, saving model to ./checkpoints/unknown_person_p9_20.h5\n",
      "31949/31949 [==============================] - 23s 710us/sample - loss: 9.2222 - val_loss: 1.3597\n",
      "Epoch 4/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 9.2316\n",
      "Epoch 4: val_loss improved from 1.35966 to 1.35065, saving model to ./checkpoints/unknown_person_p9_20.h5\n",
      "31949/31949 [==============================] - 21s 672us/sample - loss: 9.2316 - val_loss: 1.3507\n",
      "Epoch 5/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 9.1865\n",
      "Epoch 5: val_loss did not improve from 1.35065\n",
      "31949/31949 [==============================] - 22s 685us/sample - loss: 9.1865 - val_loss: 1.3668\n",
      "Epoch 6/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 9.1620\n",
      "Epoch 6: val_loss did not improve from 1.35065\n",
      "31949/31949 [==============================] - 20s 628us/sample - loss: 9.1620 - val_loss: 1.3822\n",
      "Epoch 7/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 9.1542\n",
      "Epoch 7: val_loss improved from 1.35065 to 1.34649, saving model to ./checkpoints/unknown_person_p9_20.h5\n",
      "31949/31949 [==============================] - 20s 632us/sample - loss: 9.1542 - val_loss: 1.3465\n",
      "Epoch 8/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 9.1110\n",
      "Epoch 8: val_loss did not improve from 1.34649\n",
      "31949/31949 [==============================] - 22s 700us/sample - loss: 9.1110 - val_loss: 1.3651\n",
      "Epoch 9/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 9.1044\n",
      "Epoch 9: val_loss did not improve from 1.34649\n",
      "31949/31949 [==============================] - 23s 705us/sample - loss: 9.1044 - val_loss: 1.3672\n",
      "Epoch 10/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 9.1179\n",
      "Epoch 10: val_loss did not improve from 1.34649\n",
      "31949/31949 [==============================] - 21s 669us/sample - loss: 9.1179 - val_loss: 1.3877\n",
      "Epoch 11/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 9.1099\n",
      "Epoch 11: val_loss did not improve from 1.34649\n",
      "31949/31949 [==============================] - 23s 704us/sample - loss: 9.1099 - val_loss: 1.3518\n",
      "Epoch 12/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 9.0761\n",
      "Epoch 12: val_loss did not improve from 1.34649\n",
      "31949/31949 [==============================] - 23s 705us/sample - loss: 9.0761 - val_loss: 1.3662\n",
      "Epoch 13/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 9.0522\n",
      "Epoch 13: val_loss improved from 1.34649 to 1.34285, saving model to ./checkpoints/unknown_person_p9_20.h5\n",
      "31949/31949 [==============================] - 23s 707us/sample - loss: 9.0522 - val_loss: 1.3429\n",
      "Epoch 14/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 9.0446\n",
      "Epoch 14: val_loss improved from 1.34285 to 1.34209, saving model to ./checkpoints/unknown_person_p9_20.h5\n",
      "31949/31949 [==============================] - 22s 678us/sample - loss: 9.0446 - val_loss: 1.3421\n",
      "Epoch 15/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 9.0398\n",
      "Epoch 15: val_loss did not improve from 1.34209\n",
      "31949/31949 [==============================] - 20s 628us/sample - loss: 9.0398 - val_loss: 1.3531\n",
      "Epoch 16/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 9.0238\n",
      "Epoch 16: val_loss did not improve from 1.34209\n",
      "31949/31949 [==============================] - 20s 623us/sample - loss: 9.0238 - val_loss: 1.3424\n",
      "Epoch 17/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 9.0060\n",
      "Epoch 17: val_loss did not improve from 1.34209\n",
      "31949/31949 [==============================] - 20s 630us/sample - loss: 9.0060 - val_loss: 1.3554\n",
      "Epoch 18/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 9.0025\n",
      "Epoch 18: val_loss improved from 1.34209 to 1.33785, saving model to ./checkpoints/unknown_person_p9_20.h5\n",
      "31949/31949 [==============================] - 23s 705us/sample - loss: 9.0025 - val_loss: 1.3379\n",
      "Epoch 19/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 8.9936\n",
      "Epoch 19: val_loss did not improve from 1.33785\n",
      "31949/31949 [==============================] - 23s 705us/sample - loss: 8.9936 - val_loss: 1.3471\n",
      "Epoch 20/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 9.0037\n",
      "Epoch 20: val_loss did not improve from 1.33785\n",
      "31949/31949 [==============================] - 23s 709us/sample - loss: 9.0037 - val_loss: 1.3412\n",
      "WARNING:tensorflow:Layer lstm will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_1 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_2 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_3 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_4 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_5 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_6 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_7 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_8 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_9 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_10 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_11 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_12 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_13 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_14 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_15 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_16 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_17 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_18 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_19 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_20 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_21 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_22 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_23 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_24 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_25 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_26 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_27 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_28 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_29 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_30 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_31 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_32 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_33 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_34 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_35 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_36 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 11:38:46.417091: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_16_2/lstm_cell_90/kernel/Assign' id:40598 op device:{requested: '', assigned: ''} def:{{{node lstm_16_2/lstm_cell_90/kernel/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_16_2/lstm_cell_90/kernel, lstm_16_2/lstm_cell_90/kernel/Initializer/random_uniform)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 11:38:48.896752: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_11_2/lstm_cell_85/kernel/m/Assign' id:44815 op device:{requested: '', assigned: ''} def:{{{node lstm_11_2/lstm_cell_85/kernel/m/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_11_2/lstm_cell_85/kernel/m, lstm_11_2/lstm_cell_85/kernel/m/Initializer/zeros)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Train on 31949 samples, validate on 3563 samples\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 11:38:54.116332: W tensorflow/c/c_api.cc:304] Operation '{name:'loss_5/mul' id:44579 op device:{requested: '', assigned: ''} def:{{{node loss_5/mul}} = Mul[T=DT_FLOAT, _has_manual_control_dependencies=true](loss_5/mul/x, loss_5/dense_3_loss/value)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Epoch 1/20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 11:39:07.051191: E tensorflow/core/grappler/optimizers/meta_optimizer.cc:954] layout failed: INVALID_ARGUMENT: Size of values 0 does not match size of permutation 4 @ fanin shape indropout_6/cond/dropout/SelectV2-2-TransposeNHWCToNCHW-LayoutOptimizer\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3779"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 11:39:30.871479: W tensorflow/c/c_api.cc:304] Operation '{name:'loss_5/mul' id:44579 op device:{requested: '', assigned: ''} def:{{{node loss_5/mul}} = Mul[T=DT_FLOAT, _has_manual_control_dependencies=true](loss_5/mul/x, loss_5/dense_3_loss/value)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Epoch 1: val_loss improved from inf to 1.35960, saving model to ./checkpoints/unknown_person_baseline_2_p9_20.h5\n",
      "31949/31949 [==============================] - 30s 947us/sample - loss: 1.3779 - val_loss: 1.3596\n",
      "Epoch 2/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3732\n",
      "Epoch 2: val_loss improved from 1.35960 to 1.35571, saving model to ./checkpoints/unknown_person_baseline_2_p9_20.h5\n",
      "31949/31949 [==============================] - 22s 696us/sample - loss: 1.3732 - val_loss: 1.3557\n",
      "Epoch 3/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3726\n",
      "Epoch 3: val_loss did not improve from 1.35571\n",
      "31949/31949 [==============================] - 22s 695us/sample - loss: 1.3726 - val_loss: 1.3606\n",
      "Epoch 4/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3704\n",
      "Epoch 4: val_loss improved from 1.35571 to 1.35294, saving model to ./checkpoints/unknown_person_baseline_2_p9_20.h5\n",
      "31949/31949 [==============================] - 23s 710us/sample - loss: 1.3704 - val_loss: 1.3529\n",
      "Epoch 5/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3686\n",
      "Epoch 5: val_loss improved from 1.35294 to 1.34698, saving model to ./checkpoints/unknown_person_baseline_2_p9_20.h5\n",
      "31949/31949 [==============================] - 21s 646us/sample - loss: 1.3686 - val_loss: 1.3470\n",
      "Epoch 6/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3708\n",
      "Epoch 6: val_loss did not improve from 1.34698\n",
      "31949/31949 [==============================] - 21s 657us/sample - loss: 1.3708 - val_loss: 1.3508\n",
      "Epoch 7/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3671\n",
      "Epoch 7: val_loss did not improve from 1.34698\n",
      "31949/31949 [==============================] - 20s 637us/sample - loss: 1.3671 - val_loss: 1.3690\n",
      "Epoch 8/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3685\n",
      "Epoch 8: val_loss improved from 1.34698 to 1.34595, saving model to ./checkpoints/unknown_person_baseline_2_p9_20.h5\n",
      "31949/31949 [==============================] - 22s 687us/sample - loss: 1.3685 - val_loss: 1.3460\n",
      "Epoch 9/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3657\n",
      "Epoch 9: val_loss improved from 1.34595 to 1.34313, saving model to ./checkpoints/unknown_person_baseline_2_p9_20.h5\n",
      "31949/31949 [==============================] - 20s 629us/sample - loss: 1.3657 - val_loss: 1.3431\n",
      "Epoch 10/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3643\n",
      "Epoch 10: val_loss did not improve from 1.34313\n",
      "31949/31949 [==============================] - 20s 614us/sample - loss: 1.3643 - val_loss: 1.3585\n",
      "Epoch 11/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3639\n",
      "Epoch 11: val_loss improved from 1.34313 to 1.34039, saving model to ./checkpoints/unknown_person_baseline_2_p9_20.h5\n",
      "31949/31949 [==============================] - 20s 618us/sample - loss: 1.3639 - val_loss: 1.3404\n",
      "Epoch 12/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3613\n",
      "Epoch 12: val_loss improved from 1.34039 to 1.34038, saving model to ./checkpoints/unknown_person_baseline_2_p9_20.h5\n",
      "31949/31949 [==============================] - 20s 621us/sample - loss: 1.3613 - val_loss: 1.3404\n",
      "Epoch 13/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3590\n",
      "Epoch 13: val_loss did not improve from 1.34038\n",
      "31949/31949 [==============================] - 22s 678us/sample - loss: 1.3590 - val_loss: 1.3426\n",
      "Epoch 14/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3594\n",
      "Epoch 14: val_loss did not improve from 1.34038\n",
      "31949/31949 [==============================] - 22s 689us/sample - loss: 1.3594 - val_loss: 1.3454\n",
      "Epoch 15/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3585\n",
      "Epoch 15: val_loss improved from 1.34038 to 1.34017, saving model to ./checkpoints/unknown_person_baseline_2_p9_20.h5\n",
      "31949/31949 [==============================] - 20s 619us/sample - loss: 1.3585 - val_loss: 1.3402\n",
      "Epoch 16/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3554\n",
      "Epoch 16: val_loss improved from 1.34017 to 1.33650, saving model to ./checkpoints/unknown_person_baseline_2_p9_20.h5\n",
      "31949/31949 [==============================] - 20s 617us/sample - loss: 1.3554 - val_loss: 1.3365\n",
      "Epoch 17/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3552\n",
      "Epoch 17: val_loss did not improve from 1.33650\n",
      "31949/31949 [==============================] - 20s 616us/sample - loss: 1.3552 - val_loss: 1.3381\n",
      "Epoch 18/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3562\n",
      "Epoch 18: val_loss did not improve from 1.33650\n",
      "31949/31949 [==============================] - 20s 616us/sample - loss: 1.3562 - val_loss: 1.3461\n",
      "Epoch 19/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3539\n",
      "Epoch 19: val_loss did not improve from 1.33650\n",
      "31949/31949 [==============================] - 21s 646us/sample - loss: 1.3539 - val_loss: 1.3432\n",
      "Epoch 20/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3531\n",
      "Epoch 20: val_loss did not improve from 1.33650\n",
      "31949/31949 [==============================] - 21s 650us/sample - loss: 1.3531 - val_loss: 1.3414\n",
      "Train on 31949 samples, validate on 3563 samples\n",
      "Epoch 1/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3507\n",
      "Epoch 1: val_loss improved from inf to 1.33067, saving model to ./checkpoints/unknown_person_baseline_p9_21.h5\n",
      "31949/31949 [==============================] - 20s 632us/sample - loss: 1.3507 - val_loss: 1.3307\n",
      "Epoch 2/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3517\n",
      "Epoch 2: val_loss did not improve from 1.33067\n",
      "31949/31949 [==============================] - 20s 631us/sample - loss: 1.3517 - val_loss: 1.3431\n",
      "Epoch 3/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3503\n",
      "Epoch 3: val_loss did not improve from 1.33067\n",
      "31949/31949 [==============================] - 20s 627us/sample - loss: 1.3503 - val_loss: 1.3362\n",
      "Epoch 4/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3505\n",
      "Epoch 4: val_loss improved from 1.33067 to 1.32936, saving model to ./checkpoints/unknown_person_baseline_p9_21.h5\n",
      "31949/31949 [==============================] - 20s 632us/sample - loss: 1.3505 - val_loss: 1.3294\n",
      "Epoch 5/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3464\n",
      "Epoch 5: val_loss did not improve from 1.32936\n",
      "31949/31949 [==============================] - 20s 621us/sample - loss: 1.3464 - val_loss: 1.3451\n",
      "Epoch 6/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3452\n",
      "Epoch 6: val_loss did not improve from 1.32936\n",
      "31949/31949 [==============================] - 20s 633us/sample - loss: 1.3452 - val_loss: 1.3505\n",
      "Epoch 7/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3445\n",
      "Epoch 7: val_loss did not improve from 1.32936\n",
      "31949/31949 [==============================] - 21s 653us/sample - loss: 1.3445 - val_loss: 1.3407\n",
      "Epoch 8/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3492\n",
      "Epoch 8: val_loss did not improve from 1.32936\n",
      "31949/31949 [==============================] - 20s 624us/sample - loss: 1.3492 - val_loss: 1.3508\n",
      "Epoch 9/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.4503\n",
      "Epoch 9: val_loss did not improve from 1.32936\n",
      "31949/31949 [==============================] - 20s 640us/sample - loss: 1.4503 - val_loss: 1.3428\n",
      "Epoch 10/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3544\n",
      "Epoch 10: val_loss improved from 1.32936 to 1.32926, saving model to ./checkpoints/unknown_person_baseline_p9_21.h5\n",
      "31949/31949 [==============================] - 20s 620us/sample - loss: 1.3544 - val_loss: 1.3293\n",
      "Epoch 11/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3652\n",
      "Epoch 11: val_loss did not improve from 1.32926\n",
      "31949/31949 [==============================] - 22s 677us/sample - loss: 1.3652 - val_loss: 1.3372\n",
      "Epoch 12/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3589\n",
      "Epoch 12: val_loss did not improve from 1.32926\n",
      "31949/31949 [==============================] - 22s 702us/sample - loss: 1.3589 - val_loss: 1.3424\n",
      "Epoch 13/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3682\n",
      "Epoch 13: val_loss did not improve from 1.32926\n",
      "31949/31949 [==============================] - 23s 705us/sample - loss: 1.3682 - val_loss: 1.3396\n",
      "Epoch 14/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3750\n",
      "Epoch 14: val_loss did not improve from 1.32926\n",
      "31949/31949 [==============================] - 22s 701us/sample - loss: 1.3750 - val_loss: 1.3446\n",
      "Epoch 15/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3682\n",
      "Epoch 15: val_loss did not improve from 1.32926\n",
      "31949/31949 [==============================] - 22s 681us/sample - loss: 1.3682 - val_loss: 1.3539\n",
      "Epoch 16/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3673\n",
      "Epoch 16: val_loss improved from 1.32926 to 1.32868, saving model to ./checkpoints/unknown_person_baseline_p9_21.h5\n",
      "31949/31949 [==============================] - 21s 661us/sample - loss: 1.3673 - val_loss: 1.3287\n",
      "Epoch 17/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3948\n",
      "Epoch 17: val_loss did not improve from 1.32868\n",
      "31949/31949 [==============================] - 20s 619us/sample - loss: 1.3948 - val_loss: 1.3458\n",
      "Epoch 18/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3937\n",
      "Epoch 18: val_loss did not improve from 1.32868\n",
      "31949/31949 [==============================] - 22s 675us/sample - loss: 1.3937 - val_loss: 1.3504\n",
      "Epoch 19/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.4360\n",
      "Epoch 19: val_loss did not improve from 1.32868\n",
      "31949/31949 [==============================] - 23s 706us/sample - loss: 1.4360 - val_loss: 1.3484\n",
      "Epoch 20/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.5510\n",
      "Epoch 20: val_loss did not improve from 1.32868\n",
      "31949/31949 [==============================] - 20s 637us/sample - loss: 1.5510 - val_loss: 1.4001\n",
      "Epoch 21/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.4974\n",
      "Epoch 21: val_loss did not improve from 1.32868\n",
      "31949/31949 [==============================] - 20s 636us/sample - loss: 1.4974 - val_loss: 1.3778\n",
      "Epoch 22/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.4427\n",
      "Epoch 22: val_loss did not improve from 1.32868\n",
      "31949/31949 [==============================] - 21s 649us/sample - loss: 1.4427 - val_loss: 1.3790\n",
      "Epoch 23/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.4221\n",
      "Epoch 23: val_loss did not improve from 1.32868\n",
      "31949/31949 [==============================] - 20s 618us/sample - loss: 1.4221 - val_loss: 1.3602\n",
      "Epoch 24/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.4005\n",
      "Epoch 24: val_loss did not improve from 1.32868\n",
      "31949/31949 [==============================] - 20s 617us/sample - loss: 1.4005 - val_loss: 1.3686\n",
      "Epoch 25/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3932\n",
      "Epoch 25: val_loss did not improve from 1.32868\n",
      "31949/31949 [==============================] - 20s 617us/sample - loss: 1.3932 - val_loss: 1.3560\n",
      "Epoch 26/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3839\n",
      "Epoch 26: val_loss did not improve from 1.32868\n",
      "31949/31949 [==============================] - 20s 620us/sample - loss: 1.3839 - val_loss: 1.3679\n",
      "Epoch 27/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3929\n",
      "Epoch 27: val_loss did not improve from 1.32868\n",
      "31949/31949 [==============================] - 20s 615us/sample - loss: 1.3929 - val_loss: 1.3609\n",
      "Epoch 28/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3851\n",
      "Epoch 28: val_loss did not improve from 1.32868\n",
      "31949/31949 [==============================] - 20s 615us/sample - loss: 1.3851 - val_loss: 1.3557\n",
      "Epoch 29/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3753\n",
      "Epoch 29: val_loss did not improve from 1.32868\n",
      "31949/31949 [==============================] - 21s 657us/sample - loss: 1.3753 - val_loss: 1.3668\n",
      "Epoch 30/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3793\n",
      "Epoch 30: val_loss did not improve from 1.32868\n",
      "31949/31949 [==============================] - 22s 697us/sample - loss: 1.3793 - val_loss: 1.3494\n",
      "Epoch 31/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3764\n",
      "Epoch 31: val_loss did not improve from 1.32868\n",
      "31949/31949 [==============================] - 22s 698us/sample - loss: 1.3764 - val_loss: 1.3425\n",
      "Epoch 32/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3715\n",
      "Epoch 32: val_loss did not improve from 1.32868\n",
      "31949/31949 [==============================] - 22s 693us/sample - loss: 1.3715 - val_loss: 1.3590\n",
      "Epoch 33/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3656\n",
      "Epoch 33: val_loss did not improve from 1.32868\n",
      "31949/31949 [==============================] - 20s 616us/sample - loss: 1.3656 - val_loss: 1.3472\n",
      "Epoch 34/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3637\n",
      "Epoch 34: val_loss did not improve from 1.32868\n",
      "31949/31949 [==============================] - 20s 614us/sample - loss: 1.3637 - val_loss: 1.3550\n",
      "Epoch 35/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3801\n",
      "Epoch 35: val_loss did not improve from 1.32868\n",
      "31949/31949 [==============================] - 20s 620us/sample - loss: 1.3801 - val_loss: 1.3503\n",
      "Epoch 36/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3682\n",
      "Epoch 36: val_loss did not improve from 1.32868\n",
      "31949/31949 [==============================] - 20s 617us/sample - loss: 1.3682 - val_loss: 1.3557\n",
      "Epoch 37/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3626\n",
      "Epoch 37: val_loss did not improve from 1.32868\n",
      "31949/31949 [==============================] - 22s 681us/sample - loss: 1.3626 - val_loss: 1.3503\n",
      "Epoch 38/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3598\n",
      "Epoch 38: val_loss did not improve from 1.32868\n",
      "31949/31949 [==============================] - 21s 651us/sample - loss: 1.3598 - val_loss: 1.3445\n",
      "Epoch 39/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3584\n",
      "Epoch 39: val_loss did not improve from 1.32868\n",
      "31949/31949 [==============================] - 20s 613us/sample - loss: 1.3584 - val_loss: 1.3475\n",
      "Epoch 40/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3563\n",
      "Epoch 40: val_loss did not improve from 1.32868\n",
      "31949/31949 [==============================] - 23s 706us/sample - loss: 1.3563 - val_loss: 1.3504\n",
      "Epoch 41/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3561\n",
      "Epoch 41: val_loss did not improve from 1.32868\n",
      "31949/31949 [==============================] - 22s 678us/sample - loss: 1.3561 - val_loss: 1.3533\n",
      "Epoch 42/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3545\n",
      "Epoch 42: val_loss did not improve from 1.32868\n",
      "31949/31949 [==============================] - 21s 642us/sample - loss: 1.3545 - val_loss: 1.3559\n",
      "Epoch 43/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3523\n",
      "Epoch 43: val_loss did not improve from 1.32868\n",
      "31949/31949 [==============================] - 22s 702us/sample - loss: 1.3523 - val_loss: 1.3538\n",
      "Epoch 44/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3507\n",
      "Epoch 44: val_loss did not improve from 1.32868\n",
      "31949/31949 [==============================] - 22s 700us/sample - loss: 1.3507 - val_loss: 1.3433\n",
      "Epoch 45/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3500\n",
      "Epoch 45: val_loss did not improve from 1.32868\n",
      "31949/31949 [==============================] - 22s 702us/sample - loss: 1.3500 - val_loss: 1.3370\n",
      "Epoch 46/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3497\n",
      "Epoch 46: val_loss did not improve from 1.32868\n",
      "31949/31949 [==============================] - 22s 701us/sample - loss: 1.3497 - val_loss: 1.3339\n",
      "Epoch 47/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3502\n",
      "Epoch 47: val_loss did not improve from 1.32868\n",
      "31949/31949 [==============================] - 20s 623us/sample - loss: 1.3502 - val_loss: 1.3433\n",
      "Epoch 48/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3451\n",
      "Epoch 48: val_loss did not improve from 1.32868\n",
      "31949/31949 [==============================] - 20s 619us/sample - loss: 1.3451 - val_loss: 1.3382\n",
      "Epoch 49/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3480\n",
      "Epoch 49: val_loss did not improve from 1.32868\n",
      "31949/31949 [==============================] - 20s 635us/sample - loss: 1.3480 - val_loss: 1.3346\n",
      "Epoch 50/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3423\n",
      "Epoch 50: val_loss did not improve from 1.32868\n",
      "31949/31949 [==============================] - 20s 625us/sample - loss: 1.3423 - val_loss: 1.3496\n",
      "WARNING:tensorflow:Layer lstm will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_1 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_2 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_3 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_4 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_5 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_6 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_7 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_8 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_9 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_10 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_11 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_12 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_13 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_14 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_15 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_16 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_17 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_18 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_19 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_20 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_21 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_22 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_23 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_24 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_25 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_26 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_27 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_28 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_29 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_30 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_31 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_32 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_33 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_34 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_35 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_36 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 12:03:33.651863: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_10_3/lstm_cell_121/bias/Assign' id:59027 op device:{requested: '', assigned: ''} def:{{{node lstm_10_3/lstm_cell_121/bias/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_10_3/lstm_cell_121/bias, lstm_10_3/lstm_cell_121/bias/Initializer/concat)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 12:03:36.909000: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_29_3/lstm_cell_140/kernel/v/Assign' id:65088 op device:{requested: '', assigned: ''} def:{{{node lstm_29_3/lstm_cell_140/kernel/v/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_29_3/lstm_cell_140/kernel/v, lstm_29_3/lstm_cell_140/kernel/v/Initializer/zeros)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 12:03:39.996884: W tensorflow/c/c_api.cc:304] Operation '{name:'dropout_4_3/cond/Merge' id:63349 op device:{requested: '', assigned: ''} def:{{{node dropout_4_3/cond/Merge}} = Merge[N=2, T=DT_FLOAT, _has_manual_control_dependencies=true](dropout_4_3/cond/Identity, dropout_4_3/cond/dropout/SelectV2)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1485, 1489)\n",
      "(1514, 1489)\n",
      "(1644, 1489)\n",
      "(1764, 1489)\n",
      "(1836, 1489)\n",
      "(1699, 1489)\n",
      "(1826, 1489)\n",
      "(1594, 1489)\n",
      "(1764, 1489)\n",
      "(1490, 1489)\n",
      "(1944, 1489)\n",
      "(1727, 1489)\n",
      "(1776, 1489)\n",
      "(1836, 1489)\n",
      "(1740, 1489)\n",
      "(1812, 1489)\n",
      "(946, 1489)\n",
      "(1692, 1489)\n",
      "(1860, 1489)\n",
      "{1: 6.225588500756208, 2: 8.568618578328536, 4: 10.0, 5: 4.558634190369322, 6: 4.849616407279115, 8: 9.231631458262184, 10: 6.794431502171049, 11: 7.906042957505976, 12: 9.025740962092014, 13: 6.888233643947895, 17: 9.483582353515148, 19: 8.47275271991023, 21: 8.598180221495829, 22: 1.0, 25: 7.871409840464458, 26: 6.570624359662389, 27: 6.913794091818315, 28: 5.179545787188369, 29: 2.060496962970191}\n",
      "Train on 31949 samples, validate on 3563 samples\n",
      "Epoch 1/20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 12:07:20.399436: E tensorflow/core/grappler/optimizers/meta_optimizer.cc:954] layout failed: INVALID_ARGUMENT: Size of values 0 does not match size of permutation 4 @ fanin shape indropout_7/cond/dropout/SelectV2-2-TransposeNHWCToNCHW-LayoutOptimizer\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31949/31949 [==============================] - ETA: 0s - loss: 9.7634\n",
      "Epoch 1: val_loss improved from inf to 1.39683, saving model to ./checkpoints/unknown_person_p9_21.h5\n",
      "31949/31949 [==============================] - 31s 975us/sample - loss: 9.7634 - val_loss: 1.3968\n",
      "Epoch 2/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 9.5134\n",
      "Epoch 2: val_loss improved from 1.39683 to 1.35127, saving model to ./checkpoints/unknown_person_p9_21.h5\n",
      "31949/31949 [==============================] - 21s 664us/sample - loss: 9.5134 - val_loss: 1.3513\n",
      "Epoch 3/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 9.6300\n",
      "Epoch 3: val_loss improved from 1.35127 to 1.34744, saving model to ./checkpoints/unknown_person_p9_21.h5\n",
      "31949/31949 [==============================] - 21s 657us/sample - loss: 9.6300 - val_loss: 1.3474\n",
      "Epoch 4/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 9.4775\n",
      "Epoch 4: val_loss improved from 1.34744 to 1.34218, saving model to ./checkpoints/unknown_person_p9_21.h5\n",
      "31949/31949 [==============================] - 22s 684us/sample - loss: 9.4775 - val_loss: 1.3422\n",
      "Epoch 5/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 9.4033\n",
      "Epoch 5: val_loss did not improve from 1.34218\n",
      "31949/31949 [==============================] - 22s 689us/sample - loss: 9.4033 - val_loss: 1.3529\n",
      "Epoch 6/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 9.3807\n",
      "Epoch 6: val_loss did not improve from 1.34218\n",
      "31949/31949 [==============================] - 21s 643us/sample - loss: 9.3807 - val_loss: 1.3509\n",
      "Epoch 7/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 9.3561\n",
      "Epoch 7: val_loss did not improve from 1.34218\n",
      "31949/31949 [==============================] - 22s 689us/sample - loss: 9.3561 - val_loss: 1.3676\n",
      "Epoch 8/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 9.3591\n",
      "Epoch 8: val_loss did not improve from 1.34218\n",
      "31949/31949 [==============================] - 22s 688us/sample - loss: 9.3591 - val_loss: 1.3438\n",
      "Epoch 9/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 9.3355\n",
      "Epoch 9: val_loss did not improve from 1.34218\n",
      "31949/31949 [==============================] - 20s 631us/sample - loss: 9.3355 - val_loss: 1.3586\n",
      "Epoch 10/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 9.3180\n",
      "Epoch 10: val_loss did not improve from 1.34218\n",
      "31949/31949 [==============================] - 20s 620us/sample - loss: 9.3180 - val_loss: 1.3460\n",
      "Epoch 11/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 9.3102\n",
      "Epoch 11: val_loss improved from 1.34218 to 1.33644, saving model to ./checkpoints/unknown_person_p9_21.h5\n",
      "31949/31949 [==============================] - 21s 653us/sample - loss: 9.3102 - val_loss: 1.3364\n",
      "Epoch 12/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 9.2728\n",
      "Epoch 12: val_loss did not improve from 1.33644\n",
      "31949/31949 [==============================] - 22s 676us/sample - loss: 9.2728 - val_loss: 1.3525\n",
      "Epoch 13/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 9.2679\n",
      "Epoch 13: val_loss did not improve from 1.33644\n",
      "31949/31949 [==============================] - 22s 700us/sample - loss: 9.2679 - val_loss: 1.3434\n",
      "Epoch 14/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 9.2647\n",
      "Epoch 14: val_loss did not improve from 1.33644\n",
      "31949/31949 [==============================] - 21s 644us/sample - loss: 9.2647 - val_loss: 1.3573\n",
      "Epoch 15/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 9.2601\n",
      "Epoch 15: val_loss did not improve from 1.33644\n",
      "31949/31949 [==============================] - 20s 637us/sample - loss: 9.2601 - val_loss: 1.3410\n",
      "Epoch 16/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 9.2321\n",
      "Epoch 16: val_loss improved from 1.33644 to 1.32989, saving model to ./checkpoints/unknown_person_p9_21.h5\n",
      "31949/31949 [==============================] - 23s 705us/sample - loss: 9.2321 - val_loss: 1.3299\n",
      "Epoch 17/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 9.2542\n",
      "Epoch 17: val_loss did not improve from 1.32989\n",
      "31949/31949 [==============================] - 20s 630us/sample - loss: 9.2542 - val_loss: 1.3406\n",
      "Epoch 18/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 9.2174\n",
      "Epoch 18: val_loss did not improve from 1.32989\n",
      "31949/31949 [==============================] - 20s 626us/sample - loss: 9.2174 - val_loss: 1.3455\n",
      "Epoch 19/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 9.2174\n",
      "Epoch 19: val_loss did not improve from 1.32989\n",
      "31949/31949 [==============================] - 23s 707us/sample - loss: 9.2174 - val_loss: 1.3348\n",
      "Epoch 20/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 9.2150\n",
      "Epoch 20: val_loss did not improve from 1.32989\n",
      "31949/31949 [==============================] - 23s 707us/sample - loss: 9.2150 - val_loss: 1.3304\n",
      "WARNING:tensorflow:Layer lstm will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_1 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_2 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_3 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_4 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_5 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_6 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_7 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_8 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_9 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_10 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_11 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_12 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_13 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_14 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_15 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_16 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_17 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_18 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_19 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_20 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_21 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_22 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_23 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_24 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_25 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_26 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_27 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_28 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_29 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_30 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_31 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_32 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_33 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_34 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_35 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_36 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 12:14:37.864198: W tensorflow/c/c_api.cc:304] Operation '{name:'dense_7/kernel/Assign' id:82718 op device:{requested: '', assigned: ''} def:{{{node dense_7/kernel/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](dense_7/kernel, dense_7/kernel/Initializer/random_uniform)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 12:14:41.941250: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_5_4/lstm_cell_153/recurrent_kernel/v/Assign' id:84131 op device:{requested: '', assigned: ''} def:{{{node lstm_5_4/lstm_cell_153/recurrent_kernel/v/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_5_4/lstm_cell_153/recurrent_kernel/v, lstm_5_4/lstm_cell_153/recurrent_kernel/v/Initializer/zeros)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Train on 31949 samples, validate on 3563 samples\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 12:14:48.432290: W tensorflow/c/c_api.cc:304] Operation '{name:'loss_9/mul' id:83337 op device:{requested: '', assigned: ''} def:{{{node loss_9/mul}} = Mul[T=DT_FLOAT, _has_manual_control_dependencies=true](loss_9/mul/x, loss_9/dense_3_loss/value)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Epoch 1/20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 12:15:09.684983: E tensorflow/core/grappler/optimizers/meta_optimizer.cc:954] layout failed: INVALID_ARGUMENT: Size of values 0 does not match size of permutation 4 @ fanin shape indropout_8/cond/dropout/SelectV2-2-TransposeNHWCToNCHW-LayoutOptimizer\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3998"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 12:15:31.462394: W tensorflow/c/c_api.cc:304] Operation '{name:'loss_9/mul' id:83337 op device:{requested: '', assigned: ''} def:{{{node loss_9/mul}} = Mul[T=DT_FLOAT, _has_manual_control_dependencies=true](loss_9/mul/x, loss_9/dense_3_loss/value)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Epoch 1: val_loss improved from inf to 1.34989, saving model to ./checkpoints/unknown_person_baseline_2_p9_21.h5\n",
      "31949/31949 [==============================] - 31s 963us/sample - loss: 1.3998 - val_loss: 1.3499\n",
      "Epoch 2/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3605\n",
      "Epoch 2: val_loss improved from 1.34989 to 1.34521, saving model to ./checkpoints/unknown_person_baseline_2_p9_21.h5\n",
      "31949/31949 [==============================] - 19s 598us/sample - loss: 1.3605 - val_loss: 1.3452\n",
      "Epoch 3/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3669\n",
      "Epoch 3: val_loss improved from 1.34521 to 1.33639, saving model to ./checkpoints/unknown_person_baseline_2_p9_21.h5\n",
      "31949/31949 [==============================] - 19s 599us/sample - loss: 1.3669 - val_loss: 1.3364\n",
      "Epoch 4/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3551\n",
      "Epoch 4: val_loss did not improve from 1.33639\n",
      "31949/31949 [==============================] - 19s 596us/sample - loss: 1.3551 - val_loss: 1.3486\n",
      "Epoch 5/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3499\n",
      "Epoch 5: val_loss did not improve from 1.33639\n",
      "31949/31949 [==============================] - 19s 597us/sample - loss: 1.3499 - val_loss: 1.3482\n",
      "Epoch 6/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3498\n",
      "Epoch 6: val_loss did not improve from 1.33639\n",
      "31949/31949 [==============================] - 21s 665us/sample - loss: 1.3498 - val_loss: 1.3418\n",
      "Epoch 7/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3487\n",
      "Epoch 7: val_loss did not improve from 1.33639\n",
      "31949/31949 [==============================] - 22s 693us/sample - loss: 1.3487 - val_loss: 1.3489\n",
      "Epoch 8/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3451\n",
      "Epoch 8: val_loss did not improve from 1.33639\n",
      "31949/31949 [==============================] - 19s 607us/sample - loss: 1.3451 - val_loss: 1.3396\n",
      "Epoch 9/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3455\n",
      "Epoch 9: val_loss improved from 1.33639 to 1.33497, saving model to ./checkpoints/unknown_person_baseline_2_p9_21.h5\n",
      "31949/31949 [==============================] - 21s 653us/sample - loss: 1.3455 - val_loss: 1.3350\n",
      "Epoch 10/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3423\n",
      "Epoch 10: val_loss improved from 1.33497 to 1.33014, saving model to ./checkpoints/unknown_person_baseline_2_p9_21.h5\n",
      "31949/31949 [==============================] - 20s 615us/sample - loss: 1.3423 - val_loss: 1.3301\n",
      "Epoch 11/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3415\n",
      "Epoch 11: val_loss did not improve from 1.33014\n",
      "31949/31949 [==============================] - 22s 693us/sample - loss: 1.3415 - val_loss: 1.3461\n",
      "Epoch 12/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3427\n",
      "Epoch 12: val_loss did not improve from 1.33014\n",
      "31949/31949 [==============================] - 21s 655us/sample - loss: 1.3427 - val_loss: 1.3418\n",
      "Epoch 13/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3403\n",
      "Epoch 13: val_loss did not improve from 1.33014\n",
      "31949/31949 [==============================] - 20s 639us/sample - loss: 1.3403 - val_loss: 1.3382\n",
      "Epoch 14/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3425\n",
      "Epoch 14: val_loss did not improve from 1.33014\n",
      "31949/31949 [==============================] - 21s 657us/sample - loss: 1.3425 - val_loss: 1.3325\n",
      "Epoch 15/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3375\n",
      "Epoch 15: val_loss improved from 1.33014 to 1.32974, saving model to ./checkpoints/unknown_person_baseline_2_p9_21.h5\n",
      "31949/31949 [==============================] - 20s 623us/sample - loss: 1.3375 - val_loss: 1.3297\n",
      "Epoch 16/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3371\n",
      "Epoch 16: val_loss did not improve from 1.32974\n",
      "31949/31949 [==============================] - 20s 630us/sample - loss: 1.3371 - val_loss: 1.3310\n",
      "Epoch 17/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3349\n",
      "Epoch 17: val_loss did not improve from 1.32974\n",
      "31949/31949 [==============================] - 20s 622us/sample - loss: 1.3349 - val_loss: 1.3345\n",
      "Epoch 18/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3336\n",
      "Epoch 18: val_loss did not improve from 1.32974\n",
      "31949/31949 [==============================] - 19s 604us/sample - loss: 1.3336 - val_loss: 1.3379\n",
      "Epoch 19/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3341\n",
      "Epoch 19: val_loss improved from 1.32974 to 1.32392, saving model to ./checkpoints/unknown_person_baseline_2_p9_21.h5\n",
      "31949/31949 [==============================] - 22s 691us/sample - loss: 1.3341 - val_loss: 1.3239\n",
      "Epoch 20/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3327\n",
      "Epoch 20: val_loss did not improve from 1.32392\n",
      "31949/31949 [==============================] - 21s 648us/sample - loss: 1.3327 - val_loss: 1.3287\n",
      "Train on 31949 samples, validate on 3563 samples\n",
      "Epoch 1/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3325\n",
      "Epoch 1: val_loss improved from inf to 1.32585, saving model to ./checkpoints/unknown_person_baseline_p9_22.h5\n",
      "31949/31949 [==============================] - 22s 675us/sample - loss: 1.3325 - val_loss: 1.3259\n",
      "Epoch 2/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3272\n",
      "Epoch 2: val_loss did not improve from 1.32585\n",
      "31949/31949 [==============================] - 21s 648us/sample - loss: 1.3272 - val_loss: 1.3283\n",
      "Epoch 3/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3280\n",
      "Epoch 3: val_loss improved from 1.32585 to 1.32303, saving model to ./checkpoints/unknown_person_baseline_p9_22.h5\n",
      "31949/31949 [==============================] - 22s 687us/sample - loss: 1.3280 - val_loss: 1.3230\n",
      "Epoch 4/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3274\n",
      "Epoch 4: val_loss did not improve from 1.32303\n",
      "31949/31949 [==============================] - 21s 667us/sample - loss: 1.3274 - val_loss: 1.3349\n",
      "Epoch 5/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3254\n",
      "Epoch 5: val_loss did not improve from 1.32303\n",
      "31949/31949 [==============================] - 20s 620us/sample - loss: 1.3254 - val_loss: 1.3260\n",
      "Epoch 6/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3241\n",
      "Epoch 6: val_loss did not improve from 1.32303\n",
      "31949/31949 [==============================] - 22s 683us/sample - loss: 1.3241 - val_loss: 1.3260\n",
      "Epoch 7/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3222\n",
      "Epoch 7: val_loss improved from 1.32303 to 1.31836, saving model to ./checkpoints/unknown_person_baseline_p9_22.h5\n",
      "31949/31949 [==============================] - 20s 617us/sample - loss: 1.3222 - val_loss: 1.3184\n",
      "Epoch 8/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3289\n",
      "Epoch 8: val_loss did not improve from 1.31836\n",
      "31949/31949 [==============================] - 19s 597us/sample - loss: 1.3289 - val_loss: 1.3204\n",
      "Epoch 9/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3323\n",
      "Epoch 9: val_loss did not improve from 1.31836\n",
      "31949/31949 [==============================] - 20s 614us/sample - loss: 1.3323 - val_loss: 1.3346\n",
      "Epoch 10/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3528\n",
      "Epoch 10: val_loss did not improve from 1.31836\n",
      "31949/31949 [==============================] - 20s 640us/sample - loss: 1.3528 - val_loss: 1.3372\n",
      "Epoch 11/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3495\n",
      "Epoch 11: val_loss did not improve from 1.31836\n",
      "31949/31949 [==============================] - 20s 615us/sample - loss: 1.3495 - val_loss: 1.3294\n",
      "Epoch 12/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3565\n",
      "Epoch 12: val_loss did not improve from 1.31836\n",
      "31949/31949 [==============================] - 20s 621us/sample - loss: 1.3565 - val_loss: 1.3282\n",
      "Epoch 13/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3440\n",
      "Epoch 13: val_loss did not improve from 1.31836\n",
      "31949/31949 [==============================] - 19s 610us/sample - loss: 1.3440 - val_loss: 1.3437\n",
      "Epoch 14/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3517\n",
      "Epoch 14: val_loss did not improve from 1.31836\n",
      "31949/31949 [==============================] - 19s 605us/sample - loss: 1.3517 - val_loss: 1.3302\n",
      "Epoch 15/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3750\n",
      "Epoch 15: val_loss did not improve from 1.31836\n",
      "31949/31949 [==============================] - 19s 605us/sample - loss: 1.3750 - val_loss: 1.3246\n",
      "Epoch 16/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3876\n",
      "Epoch 16: val_loss did not improve from 1.31836\n",
      "31949/31949 [==============================] - 20s 639us/sample - loss: 1.3876 - val_loss: 1.3436\n",
      "Epoch 17/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3601\n",
      "Epoch 17: val_loss did not improve from 1.31836\n",
      "31949/31949 [==============================] - 22s 701us/sample - loss: 1.3601 - val_loss: 1.3374\n",
      "Epoch 18/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3933\n",
      "Epoch 18: val_loss did not improve from 1.31836\n",
      "31949/31949 [==============================] - 22s 684us/sample - loss: 1.3933 - val_loss: 1.3400\n",
      "Epoch 19/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3902\n",
      "Epoch 19: val_loss did not improve from 1.31836\n",
      "31949/31949 [==============================] - 22s 694us/sample - loss: 1.3902 - val_loss: 1.3423\n",
      "Epoch 20/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3922\n",
      "Epoch 20: val_loss did not improve from 1.31836\n",
      "31949/31949 [==============================] - 20s 614us/sample - loss: 1.3922 - val_loss: 1.3506\n",
      "Epoch 21/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.4003\n",
      "Epoch 21: val_loss did not improve from 1.31836\n",
      "31949/31949 [==============================] - 19s 604us/sample - loss: 1.4003 - val_loss: 1.3575\n",
      "Epoch 22/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.5287\n",
      "Epoch 22: val_loss did not improve from 1.31836\n",
      "31949/31949 [==============================] - 19s 605us/sample - loss: 1.5287 - val_loss: 1.3867\n",
      "Epoch 23/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.4585\n",
      "Epoch 23: val_loss did not improve from 1.31836\n",
      "31949/31949 [==============================] - 19s 607us/sample - loss: 1.4585 - val_loss: 1.3893\n",
      "Epoch 24/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.4241\n",
      "Epoch 24: val_loss did not improve from 1.31836\n",
      "31949/31949 [==============================] - 21s 665us/sample - loss: 1.4241 - val_loss: 1.3695\n",
      "Epoch 25/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.4081\n",
      "Epoch 25: val_loss did not improve from 1.31836\n",
      "31949/31949 [==============================] - 21s 658us/sample - loss: 1.4081 - val_loss: 1.3641\n",
      "Epoch 26/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.4091\n",
      "Epoch 26: val_loss did not improve from 1.31836\n",
      "31949/31949 [==============================] - 22s 690us/sample - loss: 1.4091 - val_loss: 1.3637\n",
      "Epoch 27/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.4103\n",
      "Epoch 27: val_loss did not improve from 1.31836\n",
      "31949/31949 [==============================] - 22s 693us/sample - loss: 1.4103 - val_loss: 1.3583\n",
      "Epoch 28/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.4046\n",
      "Epoch 28: val_loss did not improve from 1.31836\n",
      "31949/31949 [==============================] - 22s 694us/sample - loss: 1.4046 - val_loss: 1.3709\n",
      "Epoch 29/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3933\n",
      "Epoch 29: val_loss did not improve from 1.31836\n",
      "31949/31949 [==============================] - 22s 692us/sample - loss: 1.3933 - val_loss: 1.3656\n",
      "Epoch 30/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3960\n",
      "Epoch 30: val_loss did not improve from 1.31836\n",
      "31949/31949 [==============================] - 22s 698us/sample - loss: 1.3960 - val_loss: 1.3608\n",
      "Epoch 31/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.4078\n",
      "Epoch 31: val_loss did not improve from 1.31836\n",
      "31949/31949 [==============================] - 22s 681us/sample - loss: 1.4078 - val_loss: 1.3605\n",
      "Epoch 32/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.4029\n",
      "Epoch 32: val_loss did not improve from 1.31836\n",
      "31949/31949 [==============================] - 19s 594us/sample - loss: 1.4029 - val_loss: 1.3713\n",
      "Epoch 33/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.4087\n",
      "Epoch 33: val_loss did not improve from 1.31836\n",
      "31949/31949 [==============================] - 21s 652us/sample - loss: 1.4087 - val_loss: 1.3611\n",
      "Epoch 34/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3995\n",
      "Epoch 34: val_loss did not improve from 1.31836\n",
      "31949/31949 [==============================] - 19s 603us/sample - loss: 1.3995 - val_loss: 1.3627\n",
      "Epoch 35/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3949\n",
      "Epoch 35: val_loss did not improve from 1.31836\n",
      "31949/31949 [==============================] - 19s 596us/sample - loss: 1.3949 - val_loss: 1.3603\n",
      "Epoch 36/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3926\n",
      "Epoch 36: val_loss did not improve from 1.31836\n",
      "31949/31949 [==============================] - 21s 665us/sample - loss: 1.3926 - val_loss: 1.3662\n",
      "Epoch 37/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3895\n",
      "Epoch 37: val_loss did not improve from 1.31836\n",
      "31949/31949 [==============================] - 21s 667us/sample - loss: 1.3895 - val_loss: 1.3597\n",
      "Epoch 38/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3923\n",
      "Epoch 38: val_loss did not improve from 1.31836\n",
      "31949/31949 [==============================] - 22s 694us/sample - loss: 1.3923 - val_loss: 1.3581\n",
      "Epoch 39/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3907\n",
      "Epoch 39: val_loss did not improve from 1.31836\n",
      "31949/31949 [==============================] - 20s 626us/sample - loss: 1.3907 - val_loss: 1.3566\n",
      "Epoch 40/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3826\n",
      "Epoch 40: val_loss did not improve from 1.31836\n",
      "31949/31949 [==============================] - 19s 600us/sample - loss: 1.3826 - val_loss: 1.3587\n",
      "Epoch 41/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3850\n",
      "Epoch 41: val_loss did not improve from 1.31836\n",
      "31949/31949 [==============================] - 19s 602us/sample - loss: 1.3850 - val_loss: 1.3520\n",
      "Epoch 42/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3841\n",
      "Epoch 42: val_loss did not improve from 1.31836\n",
      "31949/31949 [==============================] - 19s 605us/sample - loss: 1.3841 - val_loss: 1.3536\n",
      "Epoch 43/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3826\n",
      "Epoch 43: val_loss did not improve from 1.31836\n",
      "31949/31949 [==============================] - 21s 649us/sample - loss: 1.3826 - val_loss: 1.3548\n",
      "Epoch 44/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3813\n",
      "Epoch 44: val_loss did not improve from 1.31836\n",
      "31949/31949 [==============================] - 22s 675us/sample - loss: 1.3813 - val_loss: 1.3556\n",
      "Epoch 45/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3781\n",
      "Epoch 45: val_loss did not improve from 1.31836\n",
      "31949/31949 [==============================] - 22s 690us/sample - loss: 1.3781 - val_loss: 1.3612\n",
      "Epoch 46/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3772\n",
      "Epoch 46: val_loss did not improve from 1.31836\n",
      "31949/31949 [==============================] - 22s 691us/sample - loss: 1.3772 - val_loss: 1.3556\n",
      "Epoch 47/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3762\n",
      "Epoch 47: val_loss did not improve from 1.31836\n",
      "31949/31949 [==============================] - 22s 697us/sample - loss: 1.3762 - val_loss: 1.3663\n",
      "Epoch 48/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3747\n",
      "Epoch 48: val_loss did not improve from 1.31836\n",
      "31949/31949 [==============================] - 22s 694us/sample - loss: 1.3747 - val_loss: 1.3576\n",
      "Epoch 49/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3714\n",
      "Epoch 49: val_loss did not improve from 1.31836\n",
      "31949/31949 [==============================] - 21s 669us/sample - loss: 1.3714 - val_loss: 1.3483\n",
      "Epoch 50/50\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3702\n",
      "Epoch 50: val_loss did not improve from 1.31836\n",
      "31949/31949 [==============================] - 22s 692us/sample - loss: 1.3702 - val_loss: 1.3571\n",
      "WARNING:tensorflow:Layer lstm will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_1 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_2 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_3 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_4 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_5 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_6 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_7 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_8 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_9 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_10 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_11 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_12 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_13 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_14 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_15 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_16 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_17 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_18 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_19 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_20 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_21 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_22 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_23 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_24 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_25 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_26 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_27 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_28 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_29 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_30 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_31 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_32 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_33 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_34 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_35 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_36 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 12:39:27.356933: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_17_5/lstm_cell_202/bias/Assign' id:98905 op device:{requested: '', assigned: ''} def:{{{node lstm_17_5/lstm_cell_202/bias/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_17_5/lstm_cell_202/bias, lstm_17_5/lstm_cell_202/bias/Initializer/concat)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 12:39:32.379978: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_16_5/lstm_cell_201/bias/v/Assign' id:103661 op device:{requested: '', assigned: ''} def:{{{node lstm_16_5/lstm_cell_201/bias/v/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_16_5/lstm_cell_201/bias/v, lstm_16_5/lstm_cell_201/bias/v/Initializer/zeros)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 12:39:37.292592: W tensorflow/c/c_api.cc:304] Operation '{name:'dropout_4_5/cond/Merge' id:102107 op device:{requested: '', assigned: ''} def:{{{node dropout_4_5/cond/Merge}} = Merge[N=2, T=DT_FLOAT, _has_manual_control_dependencies=true](dropout_4_5/cond/Identity, dropout_4_5/cond/dropout/SelectV2)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1485, 1489)\n",
      "(1514, 1489)\n",
      "(1644, 1489)\n",
      "(1764, 1489)\n",
      "(1836, 1489)\n",
      "(1699, 1489)\n",
      "(1826, 1489)\n",
      "(1594, 1489)\n",
      "(1764, 1489)\n",
      "(1490, 1489)\n",
      "(1944, 1489)\n",
      "(1727, 1489)\n",
      "(1776, 1489)\n",
      "(1836, 1489)\n",
      "(1740, 1489)\n",
      "(1812, 1489)\n",
      "(946, 1489)\n",
      "(1692, 1489)\n",
      "(1860, 1489)\n",
      "{1: 6.109654375346415, 2: 8.265056965019856, 4: 10.0, 5: 4.960663063894436, 6: 5.080699837396725, 8: 9.194380790820441, 10: 7.138266657328572, 11: 7.90593348255271, 12: 9.287514343852528, 13: 6.953368575134533, 17: 9.570746368363126, 19: 8.638979215238756, 21: 8.824156806472947, 22: 1.0, 25: 7.897725043598023, 26: 6.394489298638296, 27: 6.710549374701516, 28: 5.3927422086855055, 29: 1.9997724631623246}\n",
      "Train on 31949 samples, validate on 3563 samples\n",
      "Epoch 1/20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 12:43:35.397885: E tensorflow/core/grappler/optimizers/meta_optimizer.cc:954] layout failed: INVALID_ARGUMENT: Size of values 0 does not match size of permutation 4 @ fanin shape indropout_9/cond/dropout/SelectV2-2-TransposeNHWCToNCHW-LayoutOptimizer\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31949/31949 [==============================] - ETA: 0s - loss: 9.5260\n",
      "Epoch 1: val_loss improved from inf to 1.35677, saving model to ./checkpoints/unknown_person_p9_22.h5\n",
      "31949/31949 [==============================] - 32s 987us/sample - loss: 9.5260 - val_loss: 1.3568\n",
      "Epoch 2/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 9.4758\n",
      "Epoch 2: val_loss improved from 1.35677 to 1.33985, saving model to ./checkpoints/unknown_person_p9_22.h5\n",
      "31949/31949 [==============================] - 20s 626us/sample - loss: 9.4758 - val_loss: 1.3398\n",
      "Epoch 3/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 9.5610\n",
      "Epoch 3: val_loss improved from 1.33985 to 1.33635, saving model to ./checkpoints/unknown_person_p9_22.h5\n",
      "31949/31949 [==============================] - 21s 668us/sample - loss: 9.5610 - val_loss: 1.3363\n",
      "Epoch 4/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 9.4137\n",
      "Epoch 4: val_loss did not improve from 1.33635\n",
      "31949/31949 [==============================] - 21s 670us/sample - loss: 9.4137 - val_loss: 1.3501\n",
      "Epoch 5/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 9.3727\n",
      "Epoch 5: val_loss improved from 1.33635 to 1.33141, saving model to ./checkpoints/unknown_person_p9_22.h5\n",
      "31949/31949 [==============================] - 22s 693us/sample - loss: 9.3727 - val_loss: 1.3314\n",
      "Epoch 6/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 9.3549\n",
      "Epoch 6: val_loss did not improve from 1.33141\n",
      "31949/31949 [==============================] - 23s 710us/sample - loss: 9.3549 - val_loss: 1.3560\n",
      "Epoch 7/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 9.3802\n",
      "Epoch 7: val_loss did not improve from 1.33141\n",
      "31949/31949 [==============================] - 23s 707us/sample - loss: 9.3802 - val_loss: 1.3431\n",
      "Epoch 8/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 9.3194\n",
      "Epoch 8: val_loss did not improve from 1.33141\n",
      "31949/31949 [==============================] - 23s 713us/sample - loss: 9.3194 - val_loss: 1.3350\n",
      "Epoch 9/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 9.3104\n",
      "Epoch 9: val_loss did not improve from 1.33141\n",
      "31949/31949 [==============================] - 22s 696us/sample - loss: 9.3104 - val_loss: 1.3322\n",
      "Epoch 10/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 9.2854\n",
      "Epoch 10: val_loss improved from 1.33141 to 1.32577, saving model to ./checkpoints/unknown_person_p9_22.h5\n",
      "31949/31949 [==============================] - 20s 641us/sample - loss: 9.2854 - val_loss: 1.3258\n",
      "Epoch 11/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 9.2739\n",
      "Epoch 11: val_loss did not improve from 1.32577\n",
      "31949/31949 [==============================] - 20s 627us/sample - loss: 9.2739 - val_loss: 1.3536\n",
      "Epoch 12/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 9.3203\n",
      "Epoch 12: val_loss did not improve from 1.32577\n",
      "31949/31949 [==============================] - 20s 625us/sample - loss: 9.3203 - val_loss: 1.3481\n",
      "Epoch 13/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 9.2627\n",
      "Epoch 13: val_loss did not improve from 1.32577\n",
      "31949/31949 [==============================] - 20s 638us/sample - loss: 9.2627 - val_loss: 1.3538\n",
      "Epoch 14/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 9.2652\n",
      "Epoch 14: val_loss improved from 1.32577 to 1.32443, saving model to ./checkpoints/unknown_person_p9_22.h5\n",
      "31949/31949 [==============================] - 22s 688us/sample - loss: 9.2652 - val_loss: 1.3244\n",
      "Epoch 15/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 9.2748\n",
      "Epoch 15: val_loss did not improve from 1.32443\n",
      "31949/31949 [==============================] - 20s 633us/sample - loss: 9.2748 - val_loss: 1.3462\n",
      "Epoch 16/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 9.2410\n",
      "Epoch 16: val_loss did not improve from 1.32443\n",
      "31949/31949 [==============================] - 23s 710us/sample - loss: 9.2410 - val_loss: 1.3349\n",
      "Epoch 17/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 9.2506\n",
      "Epoch 17: val_loss did not improve from 1.32443\n",
      "31949/31949 [==============================] - 22s 701us/sample - loss: 9.2506 - val_loss: 1.3358\n",
      "Epoch 18/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 9.2308\n",
      "Epoch 18: val_loss did not improve from 1.32443\n",
      "31949/31949 [==============================] - 22s 673us/sample - loss: 9.2308 - val_loss: 1.3412\n",
      "Epoch 19/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 9.2295\n",
      "Epoch 19: val_loss did not improve from 1.32443\n",
      "31949/31949 [==============================] - 21s 642us/sample - loss: 9.2295 - val_loss: 1.3318\n",
      "Epoch 20/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 9.2247\n",
      "Epoch 20: val_loss did not improve from 1.32443\n",
      "31949/31949 [==============================] - 22s 677us/sample - loss: 9.2247 - val_loss: 1.3289\n",
      "WARNING:tensorflow:Layer lstm will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_1 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_2 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_3 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_4 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_5 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_6 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_7 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_8 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_9 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_10 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_11 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_12 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_13 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_14 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_15 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_16 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_17 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_18 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_19 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_20 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_21 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_22 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_23 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_24 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_25 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_26 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_27 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_28 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_29 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_30 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_31 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_32 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_33 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_34 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_35 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_36 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 12:50:57.009663: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_28_6/lstm_cell_250/recurrent_kernel/Assign' id:120056 op device:{requested: '', assigned: ''} def:{{{node lstm_28_6/lstm_cell_250/recurrent_kernel/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_28_6/lstm_cell_250/recurrent_kernel, lstm_28_6/lstm_cell_250/recurrent_kernel/Initializer/mul_1)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 12:51:03.210481: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_32_6/lstm_cell_254/bias/v/Assign' id:123299 op device:{requested: '', assigned: ''} def:{{{node lstm_32_6/lstm_cell_254/bias/v/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_32_6/lstm_cell_254/bias/v, lstm_32_6/lstm_cell_254/bias/v/Initializer/zeros)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Train on 31949 samples, validate on 3563 samples\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 12:51:12.326311: W tensorflow/c/c_api.cc:304] Operation '{name:'loss_13/mul' id:122095 op device:{requested: '', assigned: ''} def:{{{node loss_13/mul}} = Mul[T=DT_FLOAT, _has_manual_control_dependencies=true](loss_13/mul/x, loss_13/dense_3_loss/value)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Epoch 1/20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 12:51:44.940863: E tensorflow/core/grappler/optimizers/meta_optimizer.cc:954] layout failed: INVALID_ARGUMENT: Size of values 0 does not match size of permutation 4 @ fanin shape indropout_10/cond/dropout/SelectV2-2-TransposeNHWCToNCHW-LayoutOptimizer\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3309"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 12:52:09.244011: W tensorflow/c/c_api.cc:304] Operation '{name:'loss_13/mul' id:122095 op device:{requested: '', assigned: ''} def:{{{node loss_13/mul}} = Mul[T=DT_FLOAT, _has_manual_control_dependencies=true](loss_13/mul/x, loss_13/dense_3_loss/value)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Epoch 1: val_loss improved from inf to 1.33542, saving model to ./checkpoints/unknown_person_baseline_2_p9_22.h5\n",
      "31949/31949 [==============================] - 37s 1ms/sample - loss: 1.3309 - val_loss: 1.3354\n",
      "Epoch 2/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3250\n",
      "Epoch 2: val_loss improved from 1.33542 to 1.32809, saving model to ./checkpoints/unknown_person_baseline_2_p9_22.h5\n",
      "31949/31949 [==============================] - 23s 716us/sample - loss: 1.3250 - val_loss: 1.3281\n",
      "Epoch 3/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3394\n",
      "Epoch 3: val_loss improved from 1.32809 to 1.32176, saving model to ./checkpoints/unknown_person_baseline_2_p9_22.h5\n",
      "31949/31949 [==============================] - 23s 714us/sample - loss: 1.3394 - val_loss: 1.3218\n",
      "Epoch 4/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3258\n",
      "Epoch 4: val_loss did not improve from 1.32176\n",
      "31949/31949 [==============================] - 23s 709us/sample - loss: 1.3258 - val_loss: 1.3364\n",
      "Epoch 5/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3203\n",
      "Epoch 5: val_loss did not improve from 1.32176\n",
      "31949/31949 [==============================] - 21s 665us/sample - loss: 1.3203 - val_loss: 1.3305\n",
      "Epoch 6/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3212\n",
      "Epoch 6: val_loss did not improve from 1.32176\n",
      "31949/31949 [==============================] - 20s 630us/sample - loss: 1.3212 - val_loss: 1.3276\n",
      "Epoch 7/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3244\n",
      "Epoch 7: val_loss did not improve from 1.32176\n",
      "31949/31949 [==============================] - 20s 628us/sample - loss: 1.3244 - val_loss: 1.3333\n",
      "Epoch 8/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3202\n",
      "Epoch 8: val_loss did not improve from 1.32176\n",
      "31949/31949 [==============================] - 20s 633us/sample - loss: 1.3202 - val_loss: 1.3229\n",
      "Epoch 9/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3187\n",
      "Epoch 9: val_loss improved from 1.32176 to 1.32023, saving model to ./checkpoints/unknown_person_baseline_2_p9_22.h5\n",
      "31949/31949 [==============================] - 20s 627us/sample - loss: 1.3187 - val_loss: 1.3202\n",
      "Epoch 10/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3200\n",
      "Epoch 10: val_loss did not improve from 1.32023\n",
      "31949/31949 [==============================] - 20s 622us/sample - loss: 1.3200 - val_loss: 1.3360\n",
      "Epoch 11/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3197\n",
      "Epoch 11: val_loss did not improve from 1.32023\n",
      "31949/31949 [==============================] - 20s 624us/sample - loss: 1.3197 - val_loss: 1.3315\n",
      "Epoch 12/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3161\n",
      "Epoch 12: val_loss did not improve from 1.32023\n",
      "31949/31949 [==============================] - 20s 628us/sample - loss: 1.3161 - val_loss: 1.3377\n",
      "Epoch 13/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3151\n",
      "Epoch 13: val_loss improved from 1.32023 to 1.31415, saving model to ./checkpoints/unknown_person_baseline_2_p9_22.h5\n",
      "31949/31949 [==============================] - 20s 628us/sample - loss: 1.3151 - val_loss: 1.3142\n",
      "Epoch 14/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3164\n",
      "Epoch 14: val_loss did not improve from 1.31415\n",
      "31949/31949 [==============================] - 22s 693us/sample - loss: 1.3164 - val_loss: 1.3271\n",
      "Epoch 15/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3152\n",
      "Epoch 15: val_loss improved from 1.31415 to 1.31188, saving model to ./checkpoints/unknown_person_baseline_2_p9_22.h5\n",
      "31949/31949 [==============================] - 23s 710us/sample - loss: 1.3152 - val_loss: 1.3119\n",
      "Epoch 16/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3147\n",
      "Epoch 16: val_loss did not improve from 1.31188\n",
      "31949/31949 [==============================] - 23s 704us/sample - loss: 1.3147 - val_loss: 1.3204\n",
      "Epoch 17/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3123\n",
      "Epoch 17: val_loss did not improve from 1.31188\n",
      "31949/31949 [==============================] - 23s 706us/sample - loss: 1.3123 - val_loss: 1.3144\n",
      "Epoch 18/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3143\n",
      "Epoch 18: val_loss did not improve from 1.31188\n",
      "31949/31949 [==============================] - 23s 716us/sample - loss: 1.3143 - val_loss: 1.3296\n",
      "Epoch 19/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3136\n",
      "Epoch 19: val_loss did not improve from 1.31188\n",
      "31949/31949 [==============================] - 22s 675us/sample - loss: 1.3136 - val_loss: 1.3357\n",
      "Epoch 20/20\n",
      "31949/31949 [==============================] - ETA: 0s - loss: 1.3143\n",
      "Epoch 20: val_loss did not improve from 1.31188\n",
      "31949/31949 [==============================] - 20s 642us/sample - loss: 1.3143 - val_loss: 1.3261\n"
     ]
    }
   ],
   "source": [
    "person_nums = [1,2,4,5,6,8,9,10,11,12,13,17,19,21,22,25,26,27,28,29]\n",
    "## Build the baseline model for emotion recognition with dropout layers\n",
    "from tensorflow.keras.models import Model\n",
    "from tensorflow.keras.layers import Input, Dense, concatenate, SimpleRNN, LSTM, Conv2D, Flatten, MaxPooling2D, GRU, AveragePooling2D, Dropout, Lambda\n",
    "from tensorflow.keras.callbacks import ModelCheckpoint\n",
    "from sklearn.preprocessing import StandardScaler\n",
    "def individual_model(features_list):\n",
    "    input_layers = []\n",
    "    hidden_layers = []\n",
    "    combined_feature = np.empty((len(features_list[0]),0))\n",
    "    for i, feature in enumerate(features_list):\n",
    "        \n",
    "        if len(feature.shape) == 3:\n",
    "            input_i = Input(shape=(feature.shape[1], feature.shape[2]))\n",
    "            input_layers.append(input_i)\n",
    "\n",
    "            hidden_i = input_i[:,:,:,None]\n",
    "            hidden_i = Conv2D(32, 3, activation='relu')(hidden_i)\n",
    "            hidden_i = AveragePooling2D((3, 3))(hidden_i)\n",
    "            hidden_i = Dropout(0.5)(hidden_i)\n",
    "            hidden_i = Conv2D(16, 3, activation='relu')(hidden_i)\n",
    "            hidden_i = AveragePooling2D((3, 3))(hidden_i)\n",
    "            hidden_i = Dropout(0.5)(hidden_i)\n",
    "            hidden_i = Conv2D(8, 1, activation='relu')(hidden_i)\n",
    "            hidden_i = AveragePooling2D((2, 2))(hidden_i)\n",
    "            hidden_i = Dropout(0.5)(hidden_i)\n",
    "            hidden_i = Conv2D(4, 1, activation='relu')(hidden_i)\n",
    "            hidden_i = AveragePooling2D((2, 2))(hidden_i)\n",
    "            hidden_i = Dropout(0.5)(hidden_i)\n",
    "            hidden_i = Flatten()(hidden_i)\n",
    "\n",
    "            hidden_layers.append(hidden_i)\n",
    "        elif feature.shape[1] <20:\n",
    "            combined_feature = np.hstack((combined_feature, feature))\n",
    "            \n",
    "        else:  # For series features\n",
    "            input_i = Input(shape=(feature.shape[1],))\n",
    "            input_layers.append(input_i)\n",
    "            hidden_i = Lambda(lambda x: x[:, :, None])(input_i)  # Add a new dimension\n",
    "            hidden_i = LSTM(4)(hidden_i)\n",
    "            hidden_layers.append(hidden_i)\n",
    "    input_i = Input(shape=(combined_feature.shape[1],))\n",
    "    input_layers.append(input_i)\n",
    "    dense_num = np.max((1, int(combined_feature.shape[1]/2)))\n",
    "    hidden_i = Dense(dense_num, activation='relu')(input_i)\n",
    "    hidden_layers.append(hidden_i)\n",
    "    print(combined_feature.shape)\n",
    "    concat_layer = concatenate(hidden_layers)\n",
    "    h = Dropout(0.2)(concat_layer)\n",
    "    h = Dense(64, activation='relu')(h)\n",
    "    h = Dense(32, activation='relu')(h)\n",
    "    output_layer = Dense(2)(h)\n",
    "    model = Model(inputs=input_layers, outputs=output_layer)\n",
    "    model.compile(loss='mean_absolute_error', optimizer='adam')\n",
    "    return model\n",
    "\n",
    "import tensorflow as tf\n",
    "from tensorflow.keras.callbacks import Callback\n",
    "import numpy as np\n",
    "\n",
    "class PruningCallback(tf.keras.callbacks.Callback):\n",
    "    def __init__(self, initial_pruning_factor=0.1, final_pruning_factor=0.5, start_epoch=0, end_epoch=None, frequency=1):\n",
    "        super(PruningCallback, self).__init__()\n",
    "        self.initial_pruning_factor = initial_pruning_factor\n",
    "        self.final_pruning_factor = final_pruning_factor\n",
    "        self.start_epoch = start_epoch\n",
    "        self.end_epoch = end_epoch if end_epoch is not None else np.inf\n",
    "        self.frequency = frequency\n",
    "        self.pruned_weights = {}\n",
    "        self.layer_importance = {}\n",
    "\n",
    "    def get_pruning_factor(self, epoch):\n",
    "        if epoch < self.start_epoch:\n",
    "            return 0\n",
    "        if epoch > self.end_epoch:\n",
    "            return self.final_pruning_factor\n",
    "        return self.initial_pruning_factor + (self.final_pruning_factor - self.initial_pruning_factor) * (epoch - self.start_epoch) / (self.end_epoch - self.start_epoch)\n",
    "\n",
    "    def on_train_begin(self, logs=None):\n",
    "        total_weight_magnitude = 0\n",
    "        for layer in self.model.layers:\n",
    "            if hasattr(layer, 'get_weights'):\n",
    "                weights = layer.get_weights()\n",
    "                layer_norm = sum(np.linalg.norm(w) for w in weights)\n",
    "                total_weight_magnitude += layer_norm\n",
    "                self.layer_importance[layer.name] = layer_norm\n",
    "    \n",
    "        # Normalize the layer importance values so they sum up to 1\n",
    "        for layer_name in self.layer_importance:\n",
    "            self.layer_importance[layer_name] /= total_weight_magnitude\n",
    "    # def on_train_begin(self, logs=None):\n",
    "    #     total_weight_magnitude = sum([np.linalg.norm(layer.get_weights()) for layer in self.model.layers if hasattr(layer, 'get_weights')])\n",
    "    #     for layer in self.model.layers:\n",
    "    #         if hasattr(layer, 'get_weights'):\n",
    "    #             self.layer_importance[layer.name] = np.linalg.norm(layer.get_weights()) / total_weight_magnitude\n",
    "\n",
    "    def get_layer_pruning_factor(self, layer_name, global_pruning_factor):\n",
    "        if layer_name in self.layer_importance:\n",
    "            importance = self.layer_importance[layer_name]\n",
    "            adjusted_pruning_factor = global_pruning_factor * (1 - importance)\n",
    "            return min(adjusted_pruning_factor, 1)  # Ensure the pruning factor is not greater than 1\n",
    "        return global_pruning_factor\n",
    "    def prune_weights(self, layer, global_pruning_factor):\n",
    "        \n",
    "        weights = layer.get_weights()\n",
    "        layer_name = layer.name\n",
    "        pruning_factor = self.get_layer_pruning_factor(layer_name, global_pruning_factor)\n",
    "\n",
    "        if layer_name not in self.pruned_weights:\n",
    "            self.pruned_weights[layer_name] = [np.zeros_like(w, dtype=bool) for w in weights]\n",
    "        \n",
    "        for i in range(len(weights)):\n",
    "            weight = weights[i]\n",
    "            # print(weight.shape)\n",
    "            # print(weight.size)\n",
    "            if weight.ndim > 1:  # Only prune dense or convolutional layers\n",
    "                unpruned_weights = np.logical_not(self.pruned_weights[layer_name][i])\n",
    "                num_unpruned = np.sum(unpruned_weights)\n",
    "                num_pruning = min(num_unpruned, int(weight.size * pruning_factor) - np.sum(self.pruned_weights[layer_name][i]))\n",
    "                num_pruning = int(weight.size * pruning_factor) - np.sum(self.pruned_weights[layer_name][i])\n",
    "                if num_pruning > 0:\n",
    "                    unpruned_flat_indices = np.flatnonzero(unpruned_weights)\n",
    "                    abs_unpruned_weights = np.abs(weight[unpruned_weights])\n",
    "                    pruning_flat_indices = np.argpartition(abs_unpruned_weights, num_pruning)[:num_pruning]\n",
    "                    \n",
    "                    indices = np.unravel_index(pruning_flat_indices, weight.shape)\n",
    "                    self.pruned_weights[layer_name][i][indices] = True\n",
    "\n",
    "                weights[i] = weights[i]*(~self.pruned_weights[layer_name][i])\n",
    "                \n",
    "        layer.set_weights(weights)\n",
    "\n",
    "    def on_epoch_end(self, epoch, logs=None):\n",
    "        if (epoch - self.start_epoch) % self.frequency != 0:\n",
    "            return\n",
    "\n",
    "        pruning_factor = self.get_pruning_factor(epoch)\n",
    "        for layer in self.model.layers:\n",
    "            if isinstance(layer, (tf.keras.layers.Dense, tf.keras.layers.Conv2D)):\n",
    "                self.prune_weights(layer, pruning_factor)\n",
    "from feature_emotion import feature_extract, split_data, label_unique_tuples, split_data_unknown\n",
    "gts, sensor_nums, walk_nums, trace_nums, people_nums, spe_centr, delta_spe_centr, spe_crest, delta_spe_crest, spe_decrease, delta_spe_decrease, spe_entropy, delta_spe_entropy, spe_flatness, delta_spe_flatness, spe_flux, delta_spe_flux, spe_kurtosis, delta_spe_kurtosis, spe_skewness, delta_spe_skewness, spe_rfp, delta_spe_rfp, spe_slope, delta_spe_slope, spe_spread, delta_spe_spread, wlk_fre, wlk_fres_trace, cwt_figs_all, cwt_sum_all_0, cwt_sum_all_1, cwt_sum_all_2, cwt_sum_all_3, high_fre_compos, pitchs, low_fre_compos, auto_corrs, real_hils, imag_hils, dur_time_1_alls, dur_time_2_alls, jitters, shimmers, jitter_rap, hrs, feature, slope, zcrs, fft_features, energy_alls, log_energy_alls, smoothe_energy_alls, legendres, double_support_time, pdps_new, lpcs, ceps_features = feature_extract(person_nums)\n",
    "\n",
    "walk_nums_all = np.squeeze(walk_nums)\n",
    "trace_nums_all = np.squeeze(trace_nums)\n",
    "people_nums_all = np.squeeze(people_nums)\n",
    "\n",
    "## 0: train, 1: validation 2: test\n",
    "\n",
    "test_person_id = [9]\n",
    "flag_tr_val_te = split_data_unknown(test_person_id, walk_nums_all, trace_nums_all, people_nums_all, person_nums, rand_seed = 42)\n",
    "## Data Normalization before training ans testing\n",
    "import tensorflow as tf\n",
    "tf.compat.v1.disable_v2_behavior()\n",
    "from tensorflow.keras.models import Model\n",
    "from tensorflow.keras.layers import Input, Dense, concatenate, LSTM\n",
    "from sklearn.preprocessing import StandardScaler\n",
    "scalers = []\n",
    "X_train_normalized = []\n",
    "X_val_normalized = []\n",
    "X_test_normalized = []\n",
    "train_idx = np.where(flag_tr_val_te ==0)[0]\n",
    "np.random.shuffle(train_idx)\n",
    "val_idx = np.where(flag_tr_val_te ==1)[0]\n",
    "test_idx = np.where(flag_tr_val_te ==2)[0]\n",
    "\n",
    "for i, feature in enumerate([spe_centr, delta_spe_centr, spe_crest, delta_spe_crest, spe_decrease, delta_spe_decrease, spe_entropy, delta_spe_entropy, spe_flatness, delta_spe_flatness, spe_flux, delta_spe_flux, spe_kurtosis, delta_spe_kurtosis, spe_skewness, delta_spe_skewness, spe_rfp, delta_spe_rfp, spe_slope, delta_spe_slope, spe_spread, delta_spe_spread, wlk_fre, wlk_fres_trace, cwt_figs_all, cwt_sum_all_0, cwt_sum_all_1, cwt_sum_all_2, cwt_sum_all_3, high_fre_compos, pitchs, low_fre_compos, auto_corrs, real_hils, imag_hils, dur_time_1_alls, dur_time_2_alls, jitters, shimmers, jitter_rap, hrs, feature, slope, zcrs, fft_features, energy_alls, log_energy_alls, smoothe_energy_alls, legendres, double_support_time, pdps_new, lpcs, ceps_features]):\n",
    "    scaler = StandardScaler()\n",
    "    if len(feature.shape)==2:\n",
    "        X_train_i = feature[train_idx,:]\n",
    "        X_val_i = feature[val_idx,:]\n",
    "        X_test_i = feature[test_idx,:]\n",
    "        X_train_normalized_i = scaler.fit_transform(X_train_i)\n",
    "        X_val_normalized_i = scaler.transform(X_val_i)\n",
    "        X_test_normalized_i = scaler.transform(X_test_i)\n",
    "        scalers.append(scaler)\n",
    "    else:\n",
    "        X_train_i = feature[train_idx,:,:]\n",
    "        X_val_i = feature[val_idx,:,:]\n",
    "        X_test_i = feature[test_idx,:,:]\n",
    "        X_train_normalized_i = scaler.fit_transform(X_train_i.reshape(X_train_i.shape[0], -1)).reshape(X_train_i.shape)\n",
    "        X_val_normalized_i = scaler.transform(X_val_i.reshape(X_val_i.shape[0], -1)).reshape(X_val_i.shape)\n",
    "        X_test_normalized_i = scaler.transform(X_test_i.reshape(X_test_i.shape[0], -1)).reshape(X_test_i.shape)\n",
    "        scalers.append(scaler)\n",
    "    X_train_normalized.append(X_train_normalized_i)\n",
    "    X_val_normalized.append(X_val_normalized_i)\n",
    "    X_test_normalized.append(X_test_normalized_i)\n",
    "y_train = gts[train_idx,:]\n",
    "y_val = gts[val_idx,:]\n",
    "y_test = gts[test_idx,:]\n",
    "X_train_normalized_new = []\n",
    "combined_feature = np.empty((len(X_train_normalized[0]),0))\n",
    "for feature in X_train_normalized:\n",
    "    if len(feature.shape) == 3:\n",
    "        X_train_normalized_new.append(feature)\n",
    "    elif feature.shape[1] <20:\n",
    "        combined_feature = np.hstack((combined_feature, feature))\n",
    "    else:\n",
    "        X_train_normalized_new.append(feature)\n",
    "X_train_normalized_new.append(combined_feature)\n",
    "\n",
    "X_val_normalized_new = []\n",
    "combined_feature = np.empty((len(X_val_normalized[0]),0))\n",
    "for feature in X_val_normalized:\n",
    "    if len(feature.shape) == 3:\n",
    "        X_val_normalized_new.append(feature)\n",
    "    elif feature.shape[1] <20:\n",
    "        combined_feature = np.hstack((combined_feature, feature))\n",
    "    else:\n",
    "        X_val_normalized_new.append(feature)\n",
    "X_val_normalized_new.append(combined_feature)\n",
    "\n",
    "X_test_normalized_new = []\n",
    "combined_feature = np.empty((len(X_test_normalized[0]),0))\n",
    "for feature in X_test_normalized:\n",
    "    if len(feature.shape) == 3:\n",
    "        X_test_normalized_new.append(feature)\n",
    "    elif feature.shape[1] <20:\n",
    "        combined_feature = np.hstack((combined_feature, feature))\n",
    "    else:\n",
    "        X_test_normalized_new.append(feature)\n",
    "X_test_normalized_new.append(combined_feature)\n",
    "\n",
    "model = individual_model(X_train_normalized)\n",
    "model.compile(loss='mean_absolute_error', optimizer='adam')\n",
    "num_epochs = 50\n",
    "\n",
    "\n",
    "# Gradually prune weights in dense and convolutional layers from 10% to 50% over the course of training, starting from epoch 0.\n",
    "\n",
    "\n",
    "rates = [0.4, 0.5, 0.6]\n",
    "idx = 20\n",
    "for r in rates:\n",
    "    model_name = './checkpoints/unknown_person_baseline_p'+str(test_person_id[0])+'_'+str(idx)+'.h5'\n",
    "    checkpoint = ModelCheckpoint(model_name, monitor='val_loss', mode='min', save_best_only=True, verbose=1)\n",
    "\n",
    "    pruning_callback = PruningCallback(initial_pruning_factor=0.01, final_pruning_factor=r, start_epoch=5, end_epoch=20, frequency=1)\n",
    "    model.fit(X_train_normalized_new, y_train, epochs=num_epochs, batch_size=1024, validation_data=(X_val_normalized_new, y_val), callbacks=[pruning_callback, checkpoint])\n",
    "    import tensorflow as tf\n",
    "    model = tf.keras.models.load_model(model_name)\n",
    "    from tensorflow.keras.models import Model\n",
    "    layers = model.layers\n",
    "    second_last_layer_output = layers[-4].output\n",
    "    feature_extractor_model = Model(inputs=model.input, outputs=second_last_layer_output)\n",
    "    train_features = feature_extractor_model.predict(X_train_normalized_new)\n",
    "    test_features = feature_extractor_model.predict(X_test_normalized_new)\n",
    "    \n",
    "    p_train = people_nums[train_idx,:]\n",
    "    p_val = people_nums[val_idx,:]\n",
    "    p_test = people_nums[test_idx,:]\n",
    "    ## Calculate the distance between test person and training person\n",
    "    def euclidean_distance(a, b):\n",
    "        return np.sqrt(np.sum((a - b) ** 2, axis=1))\n",
    "    \n",
    "    distance_dict = {}\n",
    "    for ii in range(len(person_nums)):\n",
    "        if person_nums[ii] == test_person_id[0]:\n",
    "            continue\n",
    "        else:\n",
    "            \n",
    "            ind = np.where(p_train ==person_nums[ii])[0]\n",
    "            tmp_train_features = train_features[ind, :]\n",
    "            distances = np.array([euclidean_distance(train_sample, test_features) for train_sample in tmp_train_features])\n",
    "            print(distances.shape)\n",
    "            average_distances = np.mean(distances, axis=1)\n",
    "    \n",
    "            # Step 4: Find the overall average distance\n",
    "            overall_average_distance = np.mean(average_distances)\n",
    "            distance_dict[person_nums[ii]] = overall_average_distance\n",
    "    \n",
    "    \n",
    "    def normalize_to_weights(distance_dict):\n",
    "        distances = np.array(list(distance_dict.values()))\n",
    "        # Handle the case where a distance is zero to avoid division by zero\n",
    "        distances = np.clip(distances, a_min=1e-10, a_max=None)\n",
    "        weights = 1 / distances\n",
    "        normalized_weights = weights\n",
    "        # normalized_weights = weights / sum(weights)\n",
    "        # print(sum(weights))\n",
    "        # print(sum(normalized_weights))\n",
    "        # Assign the normalized weights back to the dictionary\n",
    "        normalized_weight_dict = dict(zip(distance_dict.keys(), normalized_weights))\n",
    "        return normalized_weight_dict\n",
    "    def scale_dict_values(my_dict):\n",
    "        scaled_dict = my_dict.copy()\n",
    "        min_val = min(scaled_dict.values())\n",
    "        max_val = max(scaled_dict.values())\n",
    "        \n",
    "        for key in scaled_dict:\n",
    "            scaled_dict[key] = 1 + 9 * (scaled_dict[key] - min_val) / (max_val - min_val)\n",
    "        \n",
    "        return scaled_dict\n",
    "    weights_dict = normalize_to_weights(distance_dict)\n",
    "    weights_dict = scale_dict_values(weights_dict)\n",
    "    print(weights_dict)\n",
    "    \n",
    "    w_train = np.zeros_like(p_train)\n",
    "    for i in range(len(w_train)):\n",
    "        w_train[i] = weights_dict[int(p_train[i])]\n",
    "    \n",
    "    w_train = np.squeeze(w_train)\n",
    "    \n",
    "    import sys\n",
    "    sys.path.append('..')\n",
    "    # for layer in model.layers[-4:]:\n",
    "    #     layer.trainable = False\n",
    "    # model = individual_model.individual_model(X_train_normalized)\n",
    "    model_name = './checkpoints/unknown_person_p'+str(test_person_id[0])+'_'+str(idx)+'.h5'\n",
    "    checkpoint = ModelCheckpoint(model_name, monitor='val_loss', mode='min', save_best_only=True, verbose=1)\n",
    "    pruning_callback = PruningCallback(initial_pruning_factor=r, final_pruning_factor=r, start_epoch=1, end_epoch=20, frequency=1)\n",
    "    \n",
    "    history = model.fit(x=X_train_normalized_new, y=y_train,sample_weight= w_train,epochs=20, batch_size=1024, validation_data=(X_val_normalized_new, y_val), callbacks=[pruning_callback, checkpoint])\n",
    "    import sys\n",
    "    sys.path.append('..')\n",
    "    # for layer in model.layers[-4:]:\n",
    "    #     layer.trainable = False\n",
    "    # model = individual_model.individual_model(X_train_normalized)\n",
    "    model_name = './checkpoints/unknown_person_baseline_p'+str(test_person_id[0])+'_'+str(idx)+'.h5'\n",
    "    model = tf.keras.models.load_model(model_name)\n",
    "    model_name = './checkpoints/unknown_person_baseline_2_p'+str(test_person_id[0])+'_'+str(idx)+'.h5'\n",
    "    idx += 1\n",
    "    checkpoint = ModelCheckpoint(model_name, monitor='val_loss', mode='min', save_best_only=True, verbose=1)\n",
    "    pruning_callback = PruningCallback(initial_pruning_factor=r, final_pruning_factor=r, start_epoch=1, end_epoch=20, frequency=1)\n",
    "    \n",
    "    history = model.fit(x=X_train_normalized_new, y=y_train,epochs=20, batch_size=1024, validation_data=(X_val_normalized_new, y_val), callbacks=[pruning_callback, checkpoint])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f71b59e4-6581-4ea8-994d-a02a6654ab6b",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
